<!doctype html><html lang="zh-CN"><head><meta charset="utf-8" /><meta name="viewport" content="width=device-width,initial-scale=1" /><meta name="generator" content="VuePress 2.0.0-rc.9" /><meta name="theme" content="VuePress Theme Plume 1.0.0-rc.55" /><script id="check-dark-mode">;(function () {const um= localStorage.getItem('vuepress-theme-appearance') || 'auto';const sm = window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches;if (um === 'dark' || (um !== 'light' && sm)) {document.documentElement.classList.add('dark');}})();</script><script id="check-mac-os">document.documentElement.classList.toggle('mac', /Mac|iPhone|iPod|iPad/i.test(navigator.platform))</script><meta property="og:url" content="https://jeason.netlify.app/bioinfo/infercnv/"><meta property="og:title" content="inferCNV 推断拷贝数"><meta property="og:description" content="inferCNV使用流程 inferCNV简介 inferCNV可以被用于单细胞RNA-seq数据识别大规模的染色体拷贝数变异，如：整个染色体或是大染色体片段的获得和缺失。主要是通过与一组参考“正常”细胞相比，探索肿瘤基因组不同位置基因表达强度来实现。生成的热图可以显示每条染色体的相对表达强度。通过与正常细胞相比，很容易看出肿瘤基因组哪些区域表达过于丰..."><meta property="og:type" content="article"><meta property="og:image" content="https://cdn.jsdelivr.net/gh/Moonerss/CDN/paper/inferCNV/denoise_hard_filter.png"><meta property="og:locale" content="zh-CN"><meta property="og:updated_time" content="2024-05-13T08:48:54.000Z"><meta property="article:author" content="Jeason"><meta property="article:tag" content="biosoftware"><meta property="article:modified_time" content="2024-05-13T08:48:54.000Z"><script type="application/ld+json">{"@context":"https://schema.org","@type":"Article","headline":"inferCNV 推断拷贝数","image":["https://cdn.jsdelivr.net/gh/Moonerss/CDN/paper/inferCNV/denoise_hard_filter.png","https://cdn.jsdelivr.net/gh/Moonerss/CDN/paper/inferCNV/denoise_dynamic_sd.png","https://cdn.jsdelivr.net/gh/Moonerss/CDN/paper/inferCNV/infercnv.png"],"dateModified":"2024-05-13T08:48:54.000Z","author":[{"@type":"Person","name":"Jeason"}]}</script><title>inferCNV 推断拷贝数</title><meta name="description" content="inferCNV使用流程 inferCNV简介 inferCNV可以被用于单细胞RNA-seq数据识别大规模的染色体拷贝数变异，如：整个染色体或是大染色体片段的获得和缺失。主要是通过与一组参考“正常”细胞相比，探索肿瘤基因组不同位置基因表达强度来实现。生成的热图可以显示每条染色体的相对表达强度。通过与正常细胞相比，很容易看出肿瘤基因组哪些区域表达过于丰..."><link rel="preload" href="/assets/style-Btq4_Wsb.css" as="style"><link rel="stylesheet" href="/assets/style-Btq4_Wsb.css"><link rel="modulepreload" href="/assets/app-B6iXF0Jq.js"><link rel="modulepreload" href="/assets/index.html-Dc44YILr.js"><link rel="prefetch" href="/assets/index.html-CUrxfrQc.js" as="script"><link rel="prefetch" href="/assets/index.html-CrAAK11M.js" as="script"><link rel="prefetch" href="/assets/index.html-m1OCkFZ4.js" as="script"><link rel="prefetch" href="/assets/index.html-DpklKbiB.js" as="script"><link rel="prefetch" href="/assets/index.html-DgeLdBCj.js" as="script"><link rel="prefetch" href="/assets/index.html-DdZynjPU.js" as="script"><link rel="prefetch" href="/assets/index.html-BS80wPCd.js" as="script"><link rel="prefetch" href="/assets/index.html-BDnM9U8u.js" as="script"><link rel="prefetch" href="/assets/index.html-CS0xlF23.js" as="script"><link rel="prefetch" href="/assets/index.html-DAVdhX44.js" as="script"><link rel="prefetch" href="/assets/index.html-_b2y8g5I.js" as="script"><link rel="prefetch" href="/assets/index.html-BqX5xppv.js" as="script"><link rel="prefetch" href="/assets/index.html-Cr8adLcz.js" as="script"><link rel="prefetch" href="/assets/index.html-aGgTAbqG.js" as="script"><link rel="prefetch" href="/assets/index.html-CIx_NjWC.js" as="script"><link rel="prefetch" href="/assets/index.html-DaxJZeyl.js" as="script"><link rel="prefetch" href="/assets/index.html-BqVSKbYR.js" as="script"><link rel="prefetch" href="/assets/index.html-CAEqmjhx.js" as="script"><link rel="prefetch" href="/assets/index.html-D1TpMNoD.js" as="script"><link rel="prefetch" href="/assets/index.html-BW7qw9ey.js" as="script"><link rel="prefetch" href="/assets/index.html-C0eVu9B2.js" as="script"><link rel="prefetch" href="/assets/index.html-BpHHyRYb.js" as="script"><link rel="prefetch" href="/assets/index.html-YcWnNOR3.js" as="script"><link rel="prefetch" href="/assets/index.html-DvPeHpKr.js" as="script"><link rel="prefetch" href="/assets/index.html-GVb_j2ww.js" as="script"><link rel="prefetch" href="/assets/index.html-B2XCVCvE.js" as="script"><link rel="prefetch" href="/assets/index.html-B2wzp7sF.js" as="script"><link rel="prefetch" href="/assets/index.html-DHrdR7yz.js" as="script"><link rel="prefetch" href="/assets/index.html-D7Nph3ek.js" as="script"><link rel="prefetch" href="/assets/index.html-bO5aQHC3.js" as="script"><link rel="prefetch" href="/assets/index.html-oWXVgFcb.js" as="script"><link rel="prefetch" href="/assets/index.html-BtvxEcyn.js" as="script"><link rel="prefetch" href="/assets/index.html-SVI69W9E.js" as="script"><link rel="prefetch" href="/assets/index.html-BbxuFSWY.js" as="script"><link rel="prefetch" href="/assets/index.html-BFe9KGsX.js" as="script"><link rel="prefetch" href="/assets/index.html-CcUcGsU8.js" as="script"><link rel="prefetch" href="/assets/index.html-DuXvOCtn.js" as="script"><link rel="prefetch" href="/assets/index.html-CsYVP1Ee.js" as="script"><link rel="prefetch" href="/assets/index.html-BZNXvoee.js" as="script"><link rel="prefetch" href="/assets/index.html-DffBPlth.js" as="script"><link rel="prefetch" href="/assets/index.html-D5byNOA9.js" as="script"><link rel="prefetch" href="/assets/index.html-BIpJcaMq.js" as="script"><link rel="prefetch" href="/assets/index.html-Cclvrn0C.js" as="script"><link rel="prefetch" href="/assets/index.html-DBRMJnkP.js" as="script"><link rel="prefetch" href="/assets/index.html-C4wOU-pC.js" as="script"><link rel="prefetch" href="/assets/index.html-D46SKBHM.js" as="script"><link rel="prefetch" href="/assets/index.html-BJzbp6fj.js" as="script"><link rel="prefetch" href="/assets/index.html-BuuWC_5c.js" as="script"><link rel="prefetch" href="/assets/index.html-BdXimPXg.js" as="script"><link rel="prefetch" href="/assets/index.html-Co7EU3KZ.js" as="script"><link rel="prefetch" href="/assets/index.html-DLv0WxRB.js" as="script"><link rel="prefetch" href="/assets/index.html-T3K5oxvg.js" as="script"><link rel="prefetch" href="/assets/index.html-dRmPcSI7.js" as="script"><link rel="prefetch" href="/assets/index.html-B0QIdxhL.js" as="script"><link rel="prefetch" href="/assets/index.html-DR8uFVGu.js" as="script"><link rel="prefetch" href="/assets/index.html-BqRbC-7N.js" as="script"><link rel="prefetch" href="/assets/index.html-ByKbSJv4.js" as="script"><link rel="prefetch" href="/assets/index.html-DnflFzVH.js" as="script"><link rel="prefetch" href="/assets/index.html-CoTzCY4j.js" as="script"><link rel="prefetch" href="/assets/index.html-Bg9CM87b.js" as="script"><link rel="prefetch" href="/assets/index.html-DG8mpR12.js" as="script"><link rel="prefetch" href="/assets/index.html-BWVnAnPc.js" as="script"><link rel="prefetch" href="/assets/index.html-dPFDGAnR.js" as="script"><link rel="prefetch" href="/assets/index.html-DKU3ACQJ.js" as="script"><link rel="prefetch" href="/assets/index.html-BgklZS89.js" as="script"><link rel="prefetch" href="/assets/index.html-DcsZqy80.js" as="script"><link rel="prefetch" href="/assets/index.html-BqSD2tD7.js" as="script"><link rel="prefetch" href="/assets/index.html-DruOQWAv.js" as="script"><link rel="prefetch" href="/assets/index.html-A9a2zzOf.js" as="script"><link rel="prefetch" href="/assets/index.html-D8r7fOa3.js" as="script"><link rel="prefetch" href="/assets/index.html-BXN1kcgZ.js" as="script"><link rel="prefetch" href="/assets/index.html-6s8u9tWA.js" as="script"><link rel="prefetch" href="/assets/index.html-C1tWJLKR.js" as="script"><link rel="prefetch" href="/assets/index.html-Ck8QB2Yk.js" as="script"><link rel="prefetch" href="/assets/index.html-vvppFMQ8.js" as="script"><link rel="prefetch" href="/assets/index.html-CoTDsdMc.js" as="script"><link rel="prefetch" href="/assets/index.html-DFWqG11e.js" as="script"><link rel="prefetch" href="/assets/index.html-DBQ_Pf_a.js" as="script"><link rel="prefetch" href="/assets/index.html-CogZr12k.js" as="script"><link rel="prefetch" href="/assets/index.html-CG6pkqpU.js" as="script"><link rel="prefetch" href="/assets/index.html-Qe4RZEFP.js" as="script"><link rel="prefetch" href="/assets/index.html-B36z9cJv.js" as="script"><link rel="prefetch" href="/assets/index.html-fTMvluKk.js" as="script"><link rel="prefetch" href="/assets/index.html-DCNBrLRQ.js" as="script"><link rel="prefetch" href="/assets/index.html-CGsNPNa0.js" as="script"><link rel="prefetch" href="/assets/index.html-BqVyC19r.js" as="script"><link rel="prefetch" href="/assets/index.html-D2nf00ow.js" as="script"><link rel="prefetch" href="/assets/index.html-COInHYqu.js" as="script"><link rel="prefetch" href="/assets/index.html-Chbm4mH6.js" as="script"><link rel="prefetch" href="/assets/index.html-W2B0dRwI.js" as="script"><link rel="prefetch" href="/assets/index.html-BUlkpK18.js" as="script"><link rel="prefetch" href="/assets/index.html-Cfs9L7fI.js" as="script"><link rel="prefetch" href="/assets/index.html-GVGxHVLE.js" as="script"><link rel="prefetch" href="/assets/index.html-Ch2SyZPw.js" as="script"><link rel="prefetch" href="/assets/404.html-BQ08fjuR.js" as="script"><link rel="prefetch" href="/assets/index.html-BjVEtomp.js" as="script"><link rel="prefetch" href="/assets/index.html-I7M9H6kr.js" as="script"><link rel="prefetch" href="/assets/index.html-C9hA5HtA.js" as="script"><link rel="prefetch" href="/assets/SearchBox-BquNHWNe.js" as="script"></head><body><div id="app"><!--[--><div class="theme-plume" data-v-0f4179d5><!--[--><!--[--><span tabindex="-1" data-v-ab0f771b></span><a href="#LayoutContent" class="skip-link visually-hidden" data-v-ab0f771b> Skip to content </a><!--]--><!----><div class="nav-wrapper" data-v-0f4179d5 data-v-4aa7bc5a><div class="navbar-wrapper" data-v-4aa7bc5a data-v-f397576e><div class="container" data-v-f397576e><div class="title" data-v-f397576e><div class="navbar-title" data-v-f397576e data-v-cdd38883><a class="auto-link link title" href="/" data-v-cdd38883 data-v-66dcefcc><!--[--><!--[--><!--[--><!--[--><img class="plume-image dark logo" src="/avatar.jpeg" alt data-v-00a6d3ab><!--]--><!--[--><img class="plume-image light logo" src="/avatar.jpeg" alt data-v-00a6d3ab><!--]--><!--]--><!--]--> <!--]--><!----></a></div></div><div class="content" data-v-f397576e><div class="curtain" data-v-f397576e></div><div class="content-body" data-v-f397576e><div class="navbar-search search" data-v-f397576e><div class="search-wrapper" data-v-f7169ce8><!----><div id="local-search" data-v-f7169ce8><button type="button" class="mini-search mini-search-button" aria-label="搜索文档" data-v-f7169ce8><span class="mini-search-button-container"><svg class="mini-search-search-icon" width="20" height="20" viewBox="0 0 20 20" aria-label="search icon"><path d="M14.386 14.386l4.0877 4.0877-4.0877-4.0877c-2.9418 2.9419-7.7115 2.9419-10.6533 0-2.9419-2.9418-2.9419-7.7115 0-10.6533 2.9418-2.9419 7.7115-2.9419 10.6533 0 2.9419 2.9418 2.9419 7.7115 0 10.6533z" stroke="currentColor" fill="none" fill-rule="evenodd" stroke-linecap="round" stroke-linejoin="round"></path></svg><span class="mini-search-button-placeholder">搜索文档</span></span><span class="mini-search-button-keys"><kbd class="mini-search-button-key"></kbd><kbd class="mini-search-button-key">K</kbd></span></button></div></div></div><nav aria-labelledby="main-nav-aria-label" class="navbar-menu menu" data-v-f397576e data-v-a3f1a60e><span id="main-nav-aria-label" class="visually-hidden" data-v-a3f1a60e>Main Navigation</span><!--[--><!--[--><a class="auto-link link navbar-menu-link" href="/" data-v-a3f1a60e data-v-681c4d02 data-v-66dcefcc><!--[--><!----><i data-v-681c4d02>首页</i><!--]--><!----></a><!--]--><!--[--><a class="auto-link link navbar-menu-link" href="/NGS/filetype/" data-v-a3f1a60e data-v-681c4d02 data-v-66dcefcc><!--[--><!----><i data-v-681c4d02>NGS</i><!--]--><!----></a><!--]--><!--[--><a class="auto-link link navbar-menu-link" href="/R/" data-v-a3f1a60e data-v-681c4d02 data-v-66dcefcc><!--[--><!----><i data-v-681c4d02>R</i><!--]--><!----></a><!--]--><!--[--><a class="auto-link link navbar-menu-link" href="/py/chara_py/" data-v-a3f1a60e data-v-681c4d02 data-v-66dcefcc><!--[--><!----><i data-v-681c4d02>Python</i><!--]--><!----></a><!--]--><!--[--><div class="flyout-wrapper navbar-menu-group" data-v-a3f1a60e data-v-8ad12054><button type="button" class="button" aria-haspopup="true" aria-expanded="false" data-v-8ad12054><span class="text" data-v-8ad12054><!----><!----><span data-v-8ad12054>Analysis</span><span class="vpi-chevron-down text-icon" data-v-8ad12054></span></span></button><div class="menu" data-v-8ad12054><div class="menu-wrapper" data-v-8ad12054 data-v-a471995b><div class="items" data-v-a471995b><!--[--><!--[--><div class="menu-link" data-v-a471995b data-v-6e20746d><a class="auto-link link" href="/scRNA/" data-v-6e20746d data-v-66dcefcc><!--[--><!----><i data-v-6e20746d>scRNA</i><!--]--><!----></a></div><!--]--><!--]--></div><!--[--><!--]--></div></div></div><!--]--><!--[--><div class="flyout-wrapper navbar-menu-group active" data-v-a3f1a60e data-v-8ad12054><button type="button" class="button" aria-haspopup="true" aria-expanded="false" data-v-8ad12054><span class="text" data-v-8ad12054><!----><!----><span data-v-8ad12054>资源</span><span class="vpi-chevron-down text-icon" data-v-8ad12054></span></span></button><div class="menu" data-v-8ad12054><div class="menu-wrapper" data-v-8ad12054 data-v-a471995b><div class="items" data-v-a471995b><!--[--><!--[--><div class="menu-link" data-v-a471995b data-v-6e20746d><a class="auto-link link" href="/bioinfo/" data-v-6e20746d data-v-66dcefcc><!--[--><!----><i data-v-6e20746d>学习</i><!--]--><!----></a></div><!--]--><!--]--></div><!--[--><!--]--></div></div></div><!--]--><!--[--><div class="flyout-wrapper navbar-menu-group" data-v-a3f1a60e data-v-8ad12054><button type="button" class="button" aria-haspopup="true" aria-expanded="false" data-v-8ad12054><span class="text" data-v-8ad12054><!----><!----><span data-v-8ad12054>Awesome</span><span class="vpi-chevron-down text-icon" data-v-8ad12054></span></span></button><div class="menu" data-v-8ad12054><div class="menu-wrapper" data-v-8ad12054 data-v-a471995b><div class="items" data-v-a471995b><!--[--><!--[--><div class="menu-link" data-v-a471995b data-v-6e20746d><a class="auto-link link" href="/awesome/command/" data-v-6e20746d data-v-66dcefcc><!--[--><!----><i data-v-6e20746d>Command</i><!--]--><!----></a></div><!--]--><!--[--><div class="menu-link" data-v-a471995b data-v-6e20746d><a class="auto-link link" href="/awesome/biotools/" data-v-6e20746d data-v-66dcefcc><!--[--><!----><i data-v-6e20746d>Tools</i><!--]--><!----></a></div><!--]--><!--[--><div class="menu-link" data-v-a471995b data-v-6e20746d><a class="auto-link link" href="/awesome/Rpkgs/" data-v-6e20746d data-v-66dcefcc><!--[--><!----><i data-v-6e20746d>R package</i><!--]--><!----></a></div><!--]--><!--[--><div class="menu-link" data-v-a471995b data-v-6e20746d><a class="auto-link link" href="/awesome/Plots/" data-v-6e20746d data-v-66dcefcc><!--[--><!----><i data-v-6e20746d>Plot</i><!--]--><!----></a></div><!--]--><!--]--></div><!--[--><!--]--></div></div></div><!--]--><!--[--><a class="auto-link link navbar-menu-link" href="/blog/" data-v-a3f1a60e data-v-681c4d02 data-v-66dcefcc><!--[--><!----><i data-v-681c4d02>博客</i><!--]--><!----></a><!--]--><!--]--></nav><!----><div class="navbar-appearance appearance" data-v-f397576e data-v-2e93bdf2><button class="switch-wrapper switch-appearance" type="button" role="switch" aria-label="toggle dark mode" aria-checked="false" data-v-2e93bdf2 data-v-61e5dfef data-v-492a9bf8><span class="check" data-v-492a9bf8><span class="icon" data-v-492a9bf8><!--[--><span class="vpi-sun sun" data-v-61e5dfef></span><span class="vpi-moon moon" data-v-61e5dfef></span><!--]--></span></span></button></div><div class="social-links navbar-social-links social-links" data-v-f397576e data-v-e47592f0 data-v-82034cd4><!--[--><a class="social-link" href="https://github.com/Moonerss" aria-label="github" target="_blank" rel="noopener" data-v-82034cd4 data-v-595ca7ec><span class="vpi-social-github" /></a><!--]--></div><div class="flyout-wrapper navbar-extra extra" data-v-f397576e data-v-de8fb2b4 data-v-8ad12054><button type="button" class="button" aria-haspopup="true" aria-expanded="false" aria-label="extra navigation" data-v-8ad12054><span class="vpi-more-horizontal icon" data-v-8ad12054></span></button><div class="menu" data-v-8ad12054><div class="menu-wrapper" data-v-8ad12054 data-v-a471995b><!----><!--[--><!--[--><!----><div class="group" data-v-de8fb2b4><div class="item appearance" data-v-de8fb2b4><p class="label" data-v-de8fb2b4>外观</p><div class="appearance-action" data-v-de8fb2b4><button class="switch-wrapper switch-appearance" type="button" role="switch" aria-label="toggle dark mode" aria-checked="false" data-v-de8fb2b4 data-v-61e5dfef data-v-492a9bf8><span class="check" data-v-492a9bf8><span class="icon" data-v-492a9bf8><!--[--><span class="vpi-sun sun" data-v-61e5dfef></span><span class="vpi-moon moon" data-v-61e5dfef></span><!--]--></span></span></button></div></div></div><div class="group" data-v-de8fb2b4><div class="item social-links" data-v-de8fb2b4><div class="social-links social-links-list" data-v-de8fb2b4 data-v-82034cd4><!--[--><a class="social-link" href="https://github.com/Moonerss" aria-label="github" target="_blank" rel="noopener" data-v-82034cd4 data-v-595ca7ec><span class="vpi-social-github" /></a><!--]--></div></div></div><!--]--><!--]--></div></div></div><button type="button" class="navbar-hamburger hamburger" aria-label="mobile navigation" aria-expanded="false" aria-controls="nav-screen" data-v-f397576e data-v-1925941b><span class="container" data-v-1925941b><span class="top" data-v-1925941b></span><span class="middle" data-v-1925941b></span><span class="bottom" data-v-1925941b></span></span></button></div></div></div></div><!----></div><div class="local-nav reached-top is-blog" data-v-0f4179d5 data-v-0b456a43><button class="hidden menu" disabled aria-expanded="false" aria-controls="SidebarNav" data-v-0b456a43><span class="vpi-align-left menu-icon" data-v-0b456a43></span><span class="menu-text" data-v-0b456a43>Menu</span></button><div class="local-nav-outline-dropdown" style="--vp-vh:0px;" data-v-0b456a43 data-v-f6c5b43d><button class="" data-v-f6c5b43d>此页内容 <span class="vpi-chevron-right icon" data-v-f6c5b43d></span></button><!----></div></div><!----><div id="LayoutContent" class="layout-content" data-v-0f4179d5 data-v-bdfa9fd7><!--[--><div class="has-aside is-blog plume-page" data-v-5c470ced><div class="container" data-v-5c470ced><div class="aside" data-v-5c470ced><div class="aside-container" data-v-5c470ced><div class="aside-content" data-v-5c470ced><div class="page-aside" data-v-5c470ced data-v-f97b8b42><div class="has-outline page-aside-outline" data-v-f97b8b42><div class="content" data-v-f97b8b42><div class="outline-marker" data-v-f97b8b42></div><div class="outline-title" data-v-f97b8b42><span data-v-f97b8b42>此页内容</span><span class="vpi-print icon" data-v-f97b8b42></span></div><nav aria-labelledby="doc-outline-aria-label" data-v-f97b8b42><span id="doc-outline-aria-label" class="visually-hidden" data-v-f97b8b42> Table of Contents for current page </span><ul class="root" data-v-f97b8b42 data-v-784c5b47><!--[--><li data-v-784c5b47><a class="outline-link" href="#infercnv使用流程" data-v-784c5b47>inferCNV使用流程</a><ul class="nested" data-v-784c5b47 data-v-784c5b47><!--[--><li data-v-784c5b47><a class="outline-link" href="#infercnv简介" data-v-784c5b47>inferCNV简介</a><!----></li><li data-v-784c5b47><a class="outline-link" href="#infercnv安装" data-v-784c5b47>inferCNV安装</a><!----></li><li data-v-784c5b47><a class="outline-link" href="#infercnv运行" data-v-784c5b47>inferCNV运行</a><!----></li><li data-v-784c5b47><a class="outline-link" href="#infercnv降噪处理" data-v-784c5b47>inferCNV降噪处理</a><!----></li><li data-v-784c5b47><a class="outline-link" href="#基于hmm的cnv预测方法" data-v-784c5b47>基于HMM的CNV预测方法</a><!----></li><li data-v-784c5b47><a class="outline-link" href="#cnv-region预测报告" data-v-784c5b47>CNV region预测报告</a><!----></li><li data-v-784c5b47><a class="outline-link" href="#推断肿瘤subclusters" data-v-784c5b47>推断肿瘤subclusters</a><!----></li><li data-v-784c5b47><a class="outline-link" href="#infercnv的输出" data-v-784c5b47>inferCNV的输出</a><!----></li><li data-v-784c5b47><a class="outline-link" href="#解读infercnv图像" data-v-784c5b47>解读inferCNV图像</a><!----></li><li data-v-784c5b47><a class="outline-link" href="#参考" data-v-784c5b47>参考</a><!----></li><!--]--></ul></li><!--]--></ul></nav></div></div></div></div></div></div><div class="content" data-v-5c470ced><div class="content-container" data-v-5c470ced><!--[--><div class="page-category-wrapper" data-v-d86f3e1b><!--[--><!--[--><span class="category" data-v-d86f3e1b>bioinfo</span><!----><!--]--><!--]--></div><h1 class="page-title" data-v-d86f3e1b>inferCNV 推断拷贝数</h1><div class="page-meta-wrapper" data-v-d86f3e1b><p class="author" data-v-d86f3e1b><span class="icon vpi-user" data-v-d86f3e1b></span><span data-v-d86f3e1b>Jeason</span></p><p class="reading-time" data-v-d86f3e1b><span class="vpi-books icon" data-v-d86f3e1b></span><span data-v-d86f3e1b>1754字</span><span data-v-d86f3e1b>约6分钟</span></p><p data-v-d86f3e1b><span class="vpi-tag icon" data-v-d86f3e1b></span><!--[--><span class="tag" style="--vp-tag-color:#299764;--vp-tag-bg-color:rgba(16, 185, 129, 0.14);" data-v-d86f3e1b>biosoftware</span><!--]--></p><p class="create-time" data-v-d86f3e1b><span class="vpi-clock icon" data-v-d86f3e1b></span><span data-v-d86f3e1b>2020-01-09</span></p></div><!--]--><!--[--><div style="position:relative;" data-v-5c470ced><div class="plume-content" data-v-5c470ced><h2 id="infercnv使用流程" tabindex="-1"><a class="header-anchor" href="#infercnv使用流程"><span>inferCNV使用流程</span></a></h2><h3 id="infercnv简介" tabindex="-1"><a class="header-anchor" href="#infercnv简介"><span>inferCNV简介</span></a></h3><p><strong>inferCNV</strong>可以被用于单细胞RNA-seq数据识别大规模的染色体拷贝数变异，如：整个染色体或是大染色体片段的获得和缺失。主要是通过与一组参考“正常”细胞相比，探索肿瘤基因组不同位置基因表达强度来实现。生成的热图可以显示每条染色体的相对表达强度。通过与正常细胞相比，很容易看出肿瘤基因组哪些区域表达过于丰富或者不够丰富。</p><h3 id="infercnv安装" tabindex="-1"><a class="header-anchor" href="#infercnv安装"><span>inferCNV安装</span></a></h3><p>安装inferCNV需要同时安装<a href="https://sourceforge.net/projects/mcmc-jags/files/JAGS/4.x/" target="_blank" rel="noopener noreferrer">JAGS<span class="vpi-external-link external-icon" data-v-0cf2c236></span></a> package.</p><div class="language-r line-numbers-mode" data-ext="r" data-title="r"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span style="--shiki-light:#1E754F;--shiki-dark:#4D9375;">if</span><span style="--shiki-light:#999999;--shiki-dark:#666666;"> (</span><span style="--shiki-light:#AB5959;--shiki-dark:#CB7676;">!</span><span style="--shiki-light:#998418;--shiki-dark:#B8A965;">requireNamespace</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">(</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#B56959;--shiki-dark:#C98A7D;">BiocManager</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">, </span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">quietly</span><span style="--shiki-light:#999999;--shiki-dark:#666666;"> =</span><span style="--shiki-light:#1E754F;--shiki-dark:#4D9375;"> TRUE</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">)</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">)</span></span>
<span class="line"><span style="--shiki-light:#998418;--shiki-dark:#B8A965;">     install.packages</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">(</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#B56959;--shiki-dark:#C98A7D;">BiocManager</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">)</span></span>
<span class="line"><span style="--shiki-light:#59873A;--shiki-dark:#80A665;">BiocManager</span><span style="--shiki-light:#1E754F;--shiki-dark:#4D9375;">::</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">install</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">(</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#B56959;--shiki-dark:#C98A7D;">infercnv</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">)</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><h3 id="infercnv运行" tabindex="-1"><a class="header-anchor" href="#infercnv运行"><span>inferCNV运行</span></a></h3><p>inferCNV可以通过两步进行运行：</p><p><strong>首先需要创建一个inferCNV对象</strong>。</p><div class="language-r line-numbers-mode" data-ext="r" data-title="r"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">infercnv_obj</span><span style="--shiki-light:#999999;--shiki-dark:#666666;"> =</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;"> CreateInfercnvObject</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">(</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">raw_counts_matrix</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">=</span><span style="--shiki-light:#998418;--shiki-dark:#B8A965;">system.file</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">(</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#B56959;--shiki-dark:#C98A7D;">extdata</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">,</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;"> &quot;</span><span style="--shiki-light:#B56959;--shiki-dark:#C98A7D;">oligodendroglioma_expression_downsampled.counts.matrix.gz</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">,</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;"> package</span><span style="--shiki-light:#999999;--shiki-dark:#666666;"> =</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;"> &quot;</span><span style="--shiki-light:#B56959;--shiki-dark:#C98A7D;">infercnv</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">)</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">,</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">                                    annotations_file</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">=</span><span style="--shiki-light:#998418;--shiki-dark:#B8A965;">system.file</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">(</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#B56959;--shiki-dark:#C98A7D;">extdata</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">,            </span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#B56959;--shiki-dark:#C98A7D;">oligodendroglioma_annotations_downsampled.txt</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">, </span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">package</span><span style="--shiki-light:#999999;--shiki-dark:#666666;"> =</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;"> &quot;</span><span style="--shiki-light:#B56959;--shiki-dark:#C98A7D;">infercnv</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">),</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">                                    delim</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">=</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#A65E2B;--shiki-dark:#C99076;">\t</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">,</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">                                    gene_order_file</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">=</span><span style="--shiki-light:#998418;--shiki-dark:#B8A965;">system.file</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">(</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#B56959;--shiki-dark:#C98A7D;">extdata</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">, </span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#B56959;--shiki-dark:#C98A7D;">gencode_downsampled.EXAMPLE_ONLY_DONT_REUSE.txt</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">, </span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">package</span><span style="--shiki-light:#999999;--shiki-dark:#666666;"> =</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;"> &quot;</span><span style="--shiki-light:#B56959;--shiki-dark:#C98A7D;">infercnv</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">), </span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">ref_group_names</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">=</span><span style="--shiki-light:#998418;--shiki-dark:#B8A965;">c</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">(</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#B56959;--shiki-dark:#C98A7D;">Microglia/Macrophage</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">,</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#B56959;--shiki-dark:#C98A7D;">Oligodendrocytes (non-malignant)</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&quot;</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">))</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p>需要的输入文件有三个：</p><ul><li><p><strong>Raw Counts Matrix for Genes x Cells</strong></p><p><strong>inferCNV</strong>可以兼容来自<code>smart-seq2</code>和<code>10x genmoic</code>等单细胞测序平台的数据，count矩阵必须行是基因，列是细胞的矩阵，格式如下：</p><table><thead><tr><th></th><th>MGH54_P16_F12</th><th>MGH54_P12_C10</th><th>MGH54_P11_C11</th><th>MGH54_P15_D06</th><th>...</th></tr></thead><tbody><tr><td>A2M</td><td>0</td><td>0</td><td>0</td><td>0</td><td>...</td></tr><tr><td>AAAS</td><td>0</td><td>37</td><td>30</td><td>21</td><td>...</td></tr><tr><td>AACS</td><td>0</td><td>0</td><td>0</td><td>0</td><td>...</td></tr><tr><td>AADAT</td><td>0</td><td>0</td><td>0</td><td>0</td><td>...</td></tr><tr><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td></tr></tbody></table></li><li><p><strong>Sample annotation file</strong></p><p>样本注释文件包括两列：第一列是细胞名字，第二列是细胞的类型，以<code>tab</code>分割。如果已知正常细胞的类型，可以给出具体的细胞类型，并且，在进行计算时，每一种细胞类型都将作为单独的正常细胞存在，而不是整合成一个正常的类型。同样，如果并不能详细的获得正常细胞的类型，可以将其统称为normal，作为统一的正常细胞类型。对于恶性细胞需要标记为<code>malignant_{patient}</code>的形式。使它们在聚类时可以聚在一起。</p><div class="language-r line-numbers-mode" data-ext="r" data-title="r"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">MGH54_P2_C12</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">    Microglia</span><span style="--shiki-light:#AB5959;--shiki-dark:#CB7676;">/</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">Macrophage</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">MGH36_P6_F03</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">    Microglia</span><span style="--shiki-light:#AB5959;--shiki-dark:#CB7676;">/</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">Macrophage</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">MGH54_P16_F12</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">   Oligodendrocytes</span><span style="--shiki-light:#999999;--shiki-dark:#666666;"> (</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">non</span><span style="--shiki-light:#AB5959;--shiki-dark:#CB7676;">-</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">malignant</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">)</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">MGH54_P12_C10</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">   Oligodendrocytes</span><span style="--shiki-light:#999999;--shiki-dark:#666666;"> (</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">non</span><span style="--shiki-light:#AB5959;--shiki-dark:#CB7676;">-</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">malignant</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">)</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">MGH36_P1_B02</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">    malignant_MGH36</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">MGH36_P1_H10</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">    malignant_MGH36</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div></li><li><p><strong>Gene ordering file</strong></p><p>Gene ordering文件提供每个基因的染色体位置信息，使用<code>tab</code>分割。要分析的count矩阵中的每个基因都应该在这个基因排序文件中提供相应的基因名称和位置信息。</p><div class="language-r line-numbers-mode" data-ext="r" data-title="r"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">WASH7P</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">  chr1</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">    14363</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">   29806</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">LINC00115</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">       chr1</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">    761586</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">  762902</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">NOC2L</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">   chr1</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">    879584</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">  894689</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">MIR200A</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;"> chr1</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">    1103243</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;"> 1103332</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">SDF4</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">    chr1</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">    1152288</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;"> 1167411</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">UBE2J2</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">  chr1</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">    1189289</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;"> 1209265</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div></li></ul><blockquote><p>注意，如果没有参考的正常细胞，可以设置<code>ref_group_names=NULL</code>，使用所有细胞的平均表达作为参考基线。</p><p>此外，inferCNV默认已经去除低质量的细胞，如果要进一步进行最大或最小count值的过滤，可以设置<code>min_max_counts_per_cell=c(1e5,1e6)</code></p></blockquote><p><strong>在创建<code>infercnv_obj</code>对象之后，使用内置的<code>infercnv::run()</code>方法运行标准的infercnv过程：</strong></p><div class="language-r line-numbers-mode" data-ext="r" data-title="r"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">infercnv_obj</span><span style="--shiki-light:#999999;--shiki-dark:#666666;"> =</span><span style="--shiki-light:#59873A;--shiki-dark:#80A665;"> infercnv</span><span style="--shiki-light:#1E754F;--shiki-dark:#4D9375;">::</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">run</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">(</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">infercnv_obj</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">,</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">                             cutoff</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">=</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">1</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">,</span><span style="--shiki-light:#A0ADA0;--shiki-dark:#758575DD;"> # cutoff=1 works well for Smart-seq2, and cutoff=0.1 works well for 10x Genomics</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">                             out_dir</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">=</span><span style="--shiki-light:#998418;--shiki-dark:#B8A965;">tempfile</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">(</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">)</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">, </span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">                             cluster_by_groups</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">=</span><span style="--shiki-light:#1E754F;--shiki-dark:#4D9375;">TRUE</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">, </span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">                             denoise</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">=</span><span style="--shiki-light:#1E754F;--shiki-dark:#4D9375;">TRUE</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">,</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">                             HMM</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">=</span><span style="--shiki-light:#1E754F;--shiki-dark:#4D9375;">TRUE</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">)</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p><code>cutoff</code>决定了哪个基因可以被用于infercnv分析</p><p><code>cluster_by_groups</code>可以根据细胞注释文件对细胞进行聚类</p><h3 id="infercnv降噪处理" tabindex="-1"><a class="header-anchor" href="#infercnv降噪处理"><span>inferCNV降噪处理</span></a></h3><p>降噪处理可以降低噪声对与表达的影响，同时保留在肿瘤细胞中的表达</p><ol><li><p>使用固定阈值过滤</p><p>可以使用<code>noise_filter</code>参数设定一个特定的偏离均值的阈值</p><div class="language-r line-numbers-mode" data-ext="r" data-title="r"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">infercnv_obj</span><span style="--shiki-light:#999999;--shiki-dark:#666666;"> =</span><span style="--shiki-light:#59873A;--shiki-dark:#80A665;"> infercnv</span><span style="--shiki-light:#1E754F;--shiki-dark:#4D9375;">::</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">run</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">(</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">infercnv_obj</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">,</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">                             cutoff</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">=</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">1</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">,</span><span style="--shiki-light:#A0ADA0;--shiki-dark:#758575DD;"> # cutoff=1 works well for Smart-seq2, and cutoff=0.1 works well for 10x Genomics</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">                             out_dir</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">=</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">out_dir</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">,</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;"> </span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">                             cluster_by_groups</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">=</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">T</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">,</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;"> </span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">                             plot_steps</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">=</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">F</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">,</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">                             denoise</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">=</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">T</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">,</span><span style="--shiki-light:#A0ADA0;--shiki-dark:#758575DD;"> ## de noise</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">                             noise_filter</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">=</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">0.1</span><span style="--shiki-light:#A0ADA0;--shiki-dark:#758575DD;">   ## hard thresholds</span></span>
<span class="line"><span style="--shiki-light:#999999;--shiki-dark:#666666;">                             )</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p><img src="https://cdn.jsdelivr.net/gh/Moonerss/CDN/paper/inferCNV/denoise_hard_filter.png" alt=""></p></li><li><p>使用动态阈值过滤</p><p>默认情况下，用于去噪的硬边界是基于剩余的正常表达值的标准差来计算的。这个阈值可以使用<code>sd_amplifier</code>参数进行调整。例如，我们可以使用1.5 *标准差的过滤如下:</p><div class="language-r line-numbers-mode" data-ext="r" data-title="r"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">infercnv_obj</span><span style="--shiki-light:#999999;--shiki-dark:#666666;"> =</span><span style="--shiki-light:#59873A;--shiki-dark:#80A665;"> infercnv</span><span style="--shiki-light:#1E754F;--shiki-dark:#4D9375;">::</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">run</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">(</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">infercnv_obj</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">,</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">                             cutoff</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">=</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">1</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">,</span><span style="--shiki-light:#A0ADA0;--shiki-dark:#758575DD;"> # cutoff=1 works well for Smart-seq2, and cutoff=0.1 works well for 10x Genomics</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">                             out_dir</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">=</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">out_dir</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">,</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;"> </span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">                             cluster_by_groups</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">=</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">T</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">,</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;"> </span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">                             plot_steps</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">=</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">F</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">,</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">                             denoise</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">=</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">T</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">,</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">                             sd_amplifier</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">=</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">1.5</span><span style="--shiki-light:#A0ADA0;--shiki-dark:#758575DD;">  ## set dynamic thresholding based on the standard deviation value.</span></span>
<span class="line"><span style="--shiki-light:#999999;--shiki-dark:#666666;">                             )</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p><img src="https://cdn.jsdelivr.net/gh/Moonerss/CDN/paper/inferCNV/denoise_dynamic_sd.png" alt=""></p></li></ol><h3 id="基于hmm的cnv预测方法" tabindex="-1"><a class="header-anchor" href="#基于hmm的cnv预测方法"><span>基于HMM的CNV预测方法</span></a></h3><p>inferCNV当前支持两种基于HMM的CNV预测方法（i3和i6）。两种方法可以通过<code>infercnv::run()</code>函数中的<code>HMM_type</code>参数进行设定，默认为<code>HMM_type=&#39;i6</code></p><p>两种方法的操作对象都是对<code>CreateInfercnvObject()</code>函数计算出来的<code>infercnv</code>对象进行处理。</p><blockquote><p><a href="https://github.com/broadinstitute/infercnv/wiki/infercnv-i3-HMM-type" target="_blank" rel="noopener noreferrer">i3 HMM<span class="vpi-external-link external-icon" data-v-0cf2c236></span></a>: 包括CNV的三种状态： 拷贝数缺失（deletion）、拷贝数中立（neutral）、拷贝数扩增（amplification ）</p><p><a href="https://github.com/broadinstitute/infercnv/wiki/infercnv-i6-HMM-type" target="_blank" rel="noopener noreferrer">i6 HMM<span class="vpi-external-link external-icon" data-v-0cf2c236></span></a>: 包括更加精细定义六种CNV改变状态：</p><ul><li>0x: complete loss</li><li>0.5x: loss of one copy</li><li>1x: neutral</li><li>1.5x: addition of one copy</li><li>2x: addition of two copies</li><li>3x: essentially a placeholder for &gt;2x copies but modeled as 3x</li></ul></blockquote><h3 id="cnv-region预测报告" tabindex="-1"><a class="header-anchor" href="#cnv-region预测报告"><span>CNV region预测报告</span></a></h3><p>CNV的推断预测结果包含在以下文件夹中：</p><ul><li><p><code>HMM_CNV_predictions.*.pred_cnv_regions.dat</code> ：包含CNV区域坐标、状态分配和细胞分组的摘要信息。</p><div class="language-r line-numbers-mode" data-ext="r" data-title="r"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">cell_group_name</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">                         cnv_name</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">        state</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">   chr</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">     start</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">   end</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">malignant_MGH36.malignant_MGH36_s1</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">      chr1</span><span style="--shiki-light:#AB5959;--shiki-dark:#CB7676;">-</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">region_2</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">   0.5</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">     chr1</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">    3696784</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;"> 144612683</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">malignant_MGH36.malignant_MGH36_s1</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">      chr1</span><span style="--shiki-light:#AB5959;--shiki-dark:#CB7676;">-</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">region_4</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">   1.5</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">     chr1</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">    151336778</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">       156213123</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">malignant_MGH36.malignant_MGH36_s1</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">      chr3</span><span style="--shiki-light:#AB5959;--shiki-dark:#CB7676;">-</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">region_7</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">   1.5</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">     chr3</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">    3168600</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;"> 10285427</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">malignant_MGH36.malignant_MGH36_s1</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">      chr3</span><span style="--shiki-light:#AB5959;--shiki-dark:#CB7676;">-</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">region_9</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">   1.5</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">     chr3</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">    45429998</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">        49460186</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">malignant_MGH36.malignant_MGH36_s1</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">      chr4</span><span style="--shiki-light:#AB5959;--shiki-dark:#CB7676;">-</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">region_11</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">  0.5</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">     chr4</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">    53179</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">   187134610</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">malignant_MGH36.malignant_MGH36_s1</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">      chr5</span><span style="--shiki-light:#AB5959;--shiki-dark:#CB7676;">-</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">region_13</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">  1.5</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">     chr5</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">    134181370</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">       177037348</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div></li><li><p><code>HMM_CNV_predictions.*.cell_groupings</code>: 提供肿瘤subcluster和细胞成员的对应关系</p><div class="language-r line-numbers-mode" data-ext="r" data-title="r"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">cell_group_name</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">                         cell</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">malignant_MGH36.malignant_MGH36_s1</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">      MGH36_P3_E06</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">malignant_MGH36.malignant_MGH36_s1</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">      MGH36_P10_E07</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">malignant_MGH36.malignant_MGH36_s1</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">      MGH36_P3_C04</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">malignant_MGH36.malignant_MGH36_s1</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">      MGH36_P3_A09</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">malignant_MGH36.malignant_MGH36_s1</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">      MGH36_P2_A08</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">malignant_MGH36.malignant_MGH36_s1</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">      MGH36_P10_B08</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">malignant_MGH36.malignant_MGH36_s1</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">      MGH36_P8_H09</span></span>
<span class="line"><span style="--shiki-light:#1E754F;--shiki-dark:#4D9375;">...</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div></li><li><p><code>HMM_CNV_predictions.*.pred_cnv_genes.dat</code>: 包含每个细胞、每个基因的CNV状态分级情况</p><div class="language-r line-numbers-mode" data-ext="r" data-title="r"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">cell_group_name</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">                         gene_region_name</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;"> state</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">   gene</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">    chr</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">     start</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">   end</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">malignant_MGH36.malignant_MGH36_s1</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">      chr1</span><span style="--shiki-light:#AB5959;--shiki-dark:#CB7676;">-</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">region_2</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">   0.5</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">     DFFB</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">    chr1</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">    3696784</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;"> 3713068</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">malignant_MGH36.malignant_MGH36_s1</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">      chr1</span><span style="--shiki-light:#AB5959;--shiki-dark:#CB7676;">-</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">region_2</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">   0.5</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">     C1orf174</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">        chr1</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">    3773845</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;"> 3801993</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">malignant_MGH36.malignant_MGH36_s1</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">      chr1</span><span style="--shiki-light:#AB5959;--shiki-dark:#CB7676;">-</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">region_2</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">   0.5</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">     RPL22</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">   chr1</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">    3805689</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;"> 3816857</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">malignant_MGH36.malignant_MGH36_s1</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">      chr1</span><span style="--shiki-light:#AB5959;--shiki-dark:#CB7676;">-</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">region_2</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">   0.5</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">     ICMT</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">    chr1</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">    6241329</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;"> 6269449</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">malignant_MGH36.malignant_MGH36_s1</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">      chr1</span><span style="--shiki-light:#AB5959;--shiki-dark:#CB7676;">-</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">region_2</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">   0.5</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">     ACOT7</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">   chr1</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">    6281253</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;"> 6296032</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">malignant_MGH36.malignant_MGH36_s1</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">      chr1</span><span style="--shiki-light:#AB5959;--shiki-dark:#CB7676;">-</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">region_2</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">   0.5</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">     NOL9</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">    chr1</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">    6324329</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;"> 6454451</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">malignant_MGH36.malignant_MGH36_s1</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">      chr1</span><span style="--shiki-light:#AB5959;--shiki-dark:#CB7676;">-</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">region_2</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">   0.5</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">     KLHL21</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">  chr1</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">    6581407</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;"> 6614595</span></span>
<span class="line"><span style="--shiki-light:#1E754F;--shiki-dark:#4D9375;">...</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div></li><li><p><code>HMM_CNV_predictions.*.genes_used.dat</code>: 提供在分析中使用的基因列表</p></li></ul><h3 id="推断肿瘤subclusters" tabindex="-1"><a class="header-anchor" href="#推断肿瘤subclusters"><span>推断肿瘤subclusters</span></a></h3><p>通过设置 <code>infercnv::run(analysis_mode=&#39;subclusters&#39;)</code>, inferCNV会试着根据CNV改变的模式对细胞进行聚类。</p><p>在inferCNV中，使用的聚类方式为层次聚类（hierarchical clustering），根据层次聚类树划分不同的subcluster。有三个参数可以影响层次聚类划分的结果：</p><ul><li><code>infercnv::run(hclust_method=&#39;ward.D2&#39;)</code>：层次聚类使用的方法，支持所有的<code>hclust</code>方法。目前作者认为<code>ward.D2</code>的效果最好</li><li><code>infercnv::run(tumor_subcluster_partition_method=&#39;random_trees&#39;)</code>：划分层次聚类树的方法。可选的方法为：<code>random_trees</code>, <code>qnorm</code></li><li><code>infercnv::run(tumor_subcluster_pval=0.05)</code>:用于确定分层树中的切割点的设置</li></ul><h3 id="infercnv的输出" tabindex="-1"><a class="header-anchor" href="#infercnv的输出"><span>inferCNV的输出</span></a></h3><p>通过inferCNV的<code>run</code>函数运行后，会产生一些输出文件：</p><ul><li><code>infercnv.preliminary.png</code>:初步的inferCNV图像</li><li><code>infercnv.png </code>：采用去噪方法，生成最终的热图</li><li><code>infercnv.references.txt</code>：“正常”细胞矩阵数据值</li><li><code>infercnv.observations.txt</code>：肿瘤细胞数据值</li><li><code>infercnv.observation_groupings.txt</code>：聚类后的肿瘤细胞关系</li><li><code>infercnv.observations_dendrogram.txt </code>：肿瘤细胞聚类树状结构图</li></ul><h3 id="解读infercnv图像" tabindex="-1"><a class="header-anchor" href="#解读infercnv图像"><span>解读inferCNV图像</span></a></h3><p>通过inferCNV会得到类似于下面的聚类图：</p><p><img src="https://cdn.jsdelivr.net/gh/Moonerss/CDN/paper/inferCNV/infercnv.png" alt=""></p><p>正常细胞的表达值绘制在顶部的热图中，肿瘤细胞绘制在底部的热图中，基因从左到右排列在染色体上。将正常的细胞表达数据从肿瘤细胞表达数据中减去，产生差异，其中染色体区域扩增为红色块，染色体区域缺失为蓝色块。</p><hr><h3 id="参考" tabindex="-1"><a class="header-anchor" href="#参考"><span>参考</span></a></h3><p>https://github.com/broadinstitute/infercnv/wiki</p></div><!----></div><footer class="page-footer" data-v-5c470ced data-v-13de951c><div class="edit-info" data-v-13de951c><div class="edit-link" data-v-13de951c><a class="auto-link link edit-link-button" href="https://github.com/Moonerss/blog/edit/main/docs/bioinfo/infercnv.md" target="_blank" rel="noreferrer" data-v-13de951c data-v-66dcefcc><!--[--><span class="vpi-square-pen edit-link-icon" aria-label="edit icon" data-v-13de951c></span> 在 GitHub 上编辑此页<!--]--><!----></a></div><!----></div><!----><nav class="prev-next" data-v-13de951c><div class="pager" data-v-13de951c><a class="auto-link link pager-link prev" href="/software/transmission/" data-v-13de951c data-v-66dcefcc><!--[--><span class="desc" data-v-13de951c>上一页</span><span class="title" data-v-13de951c>服务器互传文件</span><!--]--><!----></a></div><div class="pager" data-v-13de951c><a class="auto-link link pager-link next" href="/bioinfo/pheatmap/" data-v-13de951c data-v-66dcefcc><!--[--><span class="desc" data-v-13de951c>下一页</span><span class="title" data-v-13de951c>pheatmap使用指南</span><!--]--><!----></a></div></nav></footer><!----><!--]--></div></div></div></div><button style="display:none;" type="button" class="back-to-top-button" aria-label="back to top" data-v-0f4179d5 data-v-0af269ee><span class="percent" data-v-0af269ee>0%</span><span class="show icon vpi-back-to-top" data-v-0af269ee></span><svg aria-hidden="true" data-v-0af269ee><circle cx="50%" cy="50%" style="stroke-dasharray:calc(0% - 12.566370614359172px) calc(314.1592653589793% - 12.566370614359172px);" data-v-0af269ee></circle></svg></button><footer class="plume-footer" data-v-0f4179d5 data-v-24205ac6><div class="container" data-v-24205ac6><p class="message" data-v-24205ac6>Power by <a target="_blank" href="https://v2.vuepress.vuejs.org/">VuePress</a> & <a target="_blank" href="https://github.com/pengzhanbo/vuepress-theme-plume">vuepress-theme-plume</a></p><p class="copyright" data-v-24205ac6>Copyright © 2024-present Jeason</p></div></footer><!--]--></div><!--]--></div><!--[--><!--]--><!--]--></div><script type="module" src="/assets/app-B6iXF0Jq.js" defer></script></body></html>