---
title: Awesome bioinfomatics tools
author: Jeason
createTime: 2024/05/10 21:57:24
permalink: /awesome/biotools/
---

::: info
收集方便使用的生物信息学软件
:::

## gtf
+ [GffRead](https://github.com/gpertea/gffread): GFF/GTF utility providing format conversions, region filtering, FASTA sequence extraction and more  

## genome

+ [genomepy](https://github.com/vanheeringen-lab/genomepy) genomepy is designed to provide a simple and straightforward way to download and use genomic data. This includes (1) searching available data, (2) showing the available metadata, (3) automatically downloading, preprocessing and matching data and (4) generating optional aligner indexes. All with sensible, yet controllable defaults. Currently, genomepy supports Ensembl, UCSC, NCBI and GENCODE.  
+ [mosdepth](https://github.com/brentp/mosdepth) 快速计算测序深度  
+ 

## Annotation

+ [eggNOG-mapper](https://github.com/eggnogdb/eggnog-mapper) Fast genome-wide functional annotation through orthology assignment  

## Single cell

### Analysis toolkit  

+ [singleCellTK](https://github.com/compbiomed/singleCellTK) R interface to several popular single-cell RNA-sequencing (scRNAseq) data preprocessing, quality control, analysis, and visualization tools.  
+ [SnapATAC2](https://github.com/kaizhang/SnapATAC2) A Python/Rust package for single-cell epigenomics analysis  
+ [scDIOR](https://github.com/JiekaiLab/scDIOR) single-cell data transformation between platforms of R and Python based on Hierarchical Data Format Version 5  
+ [cNMF](https://github.com/dylkot/cNMF) inferring gene expression programs from scRNA-Seq  
+ [SCEVAN](https://github.com/AntonioDeFalco/SCEVAN) R package that automatically classifies the cells in the scRNA data by segregating non-malignant cells of tumor microenviroment from the malignant cells. It also infers the copy number profile of malignant cells, identifies subclonal structures and analyses the specific and shared alterations of each subpopulation.  
+ [Scanorama](https://github.com/brianhie/scanorama) batch-correction and integration of heterogeneous scRNA-seq datasets  
+ [cellqc](https://github.com/lijinbio/cellqc) standardized quality control pipeline of single-cell RNA-Seq data  


### Annotation cell type  

+ [scCATCH](https://github.com/ZJUFanLab/scCATCH) Automatic Annotation on Cell Types of Clusters from Single-Cell RNA Sequencing Data  

### Metabolism  

+ [Compass](https://github.com/YosefLab/Compass) In-Silico Modeling of Metabolic Heterogeneity using Single-Cell Transcriptomes  


### Spatial omics  

+ [Giotto Suite](https://github.com/drieslab/Giotto) Spatial omics analysis toolbox  
+ [Redeconve](https://github.com/ZxZhou4150/Redeconve) 基于单细胞空间组学参考的去卷积方法  

## CHIP-seq & ATAC-seq  

+ [SICER](https://github.com/zanglab/SICER2) 针对 broad peak 的识别工具  


## Machine learning  

+ [Lazy Predict](https://github.com/shankarpandala/lazypredict) Lazy Predict help build a lot of basic models without much code and helps understand which models works better without any parameter tuning  

## Analyis  

+ [GSEApy](https://github.com/zqfang/GSEApy) Gene Set Enrichment Analysis in Python.  
+ [CPTAC](https://github.com/PayneLab/cptac) Python packaging for CPTAC data  


## Visualization  

+ [CORAL](https://github.com/dphansti/CORAL) 绘制标记颜色激酶树的shiny app
+ [Chromoscope](https://github.com/hms-dbmi/chromoscope) 人类基因组结构变异的交互式多尺度可视化  
+ [SparK](https://github.com/harbourlab/SparK) 绘制出版级别的track plot  
+ [shinyCircos](https://venyao.xyz/shinycircos/) 在线绘制Ciros图的shiny app  
