<!doctype html><html lang="zh-CN"><head><meta charset="utf-8" /><meta name="viewport" content="width=device-width,initial-scale=1" /><meta name="generator" content="VuePress 2.0.0-rc.9" /><meta name="theme" content="VuePress Theme Plume 1.0.0-rc.55" /><script id="check-dark-mode">;(function () {const um= localStorage.getItem('vuepress-theme-appearance') || 'auto';const sm = window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches;if (um === 'dark' || (um !== 'light' && sm)) {document.documentElement.classList.add('dark');}})();</script><script id="check-mac-os">document.documentElement.classList.toggle('mac', /Mac|iPhone|iPod|iPad/i.test(navigator.platform))</script><meta property="og:url" content="https://jeason.netlify.app/NGS/filetype/"><meta property="og:title" content="测序常见文件格式"><meta property="og:description" content="生物信息学常见文件格式 FASTQ FASTQ用于保存生物序列（通常是核酸序列）和其测序质量信息的标准格式。 其序列以及质量信息都是使用一个ASCII字符标示，最初由Sanger开发。 目的是将FASTA序列与质量数据放到一起，目前已经成为高通量测序结果的实施标准。 FASTQ文件中每个序列通常有四行： 第一行是序列标识以及相关的描述信息，以‘@’开头..."><meta property="og:type" content="article"><meta property="og:image" content="https://cdn.jsdelivr.net/gh/Moonerss/CDN/paper/filtype/wiki-fastq-format.png"><meta property="og:locale" content="zh-CN"><meta property="og:updated_time" content="2024-05-13T13:59:34.000Z"><meta property="article:author" content="Jeason"><meta property="article:modified_time" content="2024-05-13T13:59:34.000Z"><script type="application/ld+json">{"@context":"https://schema.org","@type":"Article","headline":"测序常见文件格式","image":["https://cdn.jsdelivr.net/gh/Moonerss/CDN/paper/filtype/wiki-fastq-format.png","https://cdn.jsdelivr.net/gh/Moonerss/CDN/paper/filtype/vcf_variation_class.jpg","https://cdn.jsdelivr.net/gh/Moonerss/CDN/paper/filtype/vcf_file_format.jpg"],"dateModified":"2024-05-13T13:59:34.000Z","author":[{"@type":"Person","name":"Jeason"}]}</script><title>测序常见文件格式</title><meta name="description" content="生物信息学常见文件格式 FASTQ FASTQ用于保存生物序列（通常是核酸序列）和其测序质量信息的标准格式。 其序列以及质量信息都是使用一个ASCII字符标示，最初由Sanger开发。 目的是将FASTA序列与质量数据放到一起，目前已经成为高通量测序结果的实施标准。 FASTQ文件中每个序列通常有四行： 第一行是序列标识以及相关的描述信息，以‘@’开头..."><link rel="preload" href="/assets/style-Btq4_Wsb.css" as="style"><link rel="stylesheet" href="/assets/style-Btq4_Wsb.css"><link rel="modulepreload" href="/assets/app-B6iXF0Jq.js"><link rel="modulepreload" href="/assets/index.html-ByKbSJv4.js"><link rel="prefetch" href="/assets/index.html-CUrxfrQc.js" as="script"><link rel="prefetch" href="/assets/index.html-CrAAK11M.js" as="script"><link rel="prefetch" href="/assets/index.html-m1OCkFZ4.js" as="script"><link rel="prefetch" href="/assets/index.html-DpklKbiB.js" as="script"><link rel="prefetch" href="/assets/index.html-DgeLdBCj.js" as="script"><link rel="prefetch" href="/assets/index.html-DdZynjPU.js" as="script"><link rel="prefetch" href="/assets/index.html-BS80wPCd.js" as="script"><link rel="prefetch" href="/assets/index.html-BDnM9U8u.js" as="script"><link rel="prefetch" href="/assets/index.html-CS0xlF23.js" as="script"><link rel="prefetch" href="/assets/index.html-Dc44YILr.js" as="script"><link rel="prefetch" href="/assets/index.html-DAVdhX44.js" as="script"><link rel="prefetch" href="/assets/index.html-_b2y8g5I.js" as="script"><link rel="prefetch" href="/assets/index.html-BqX5xppv.js" as="script"><link rel="prefetch" href="/assets/index.html-Cr8adLcz.js" as="script"><link rel="prefetch" href="/assets/index.html-aGgTAbqG.js" as="script"><link rel="prefetch" href="/assets/index.html-CIx_NjWC.js" as="script"><link rel="prefetch" href="/assets/index.html-DaxJZeyl.js" as="script"><link rel="prefetch" href="/assets/index.html-BqVSKbYR.js" as="script"><link rel="prefetch" href="/assets/index.html-CAEqmjhx.js" as="script"><link rel="prefetch" href="/assets/index.html-D1TpMNoD.js" as="script"><link rel="prefetch" href="/assets/index.html-BW7qw9ey.js" as="script"><link rel="prefetch" href="/assets/index.html-C0eVu9B2.js" as="script"><link rel="prefetch" href="/assets/index.html-BpHHyRYb.js" as="script"><link rel="prefetch" href="/assets/index.html-YcWnNOR3.js" as="script"><link rel="prefetch" href="/assets/index.html-DvPeHpKr.js" as="script"><link rel="prefetch" href="/assets/index.html-GVb_j2ww.js" as="script"><link rel="prefetch" href="/assets/index.html-B2XCVCvE.js" as="script"><link rel="prefetch" href="/assets/index.html-B2wzp7sF.js" as="script"><link rel="prefetch" href="/assets/index.html-DHrdR7yz.js" as="script"><link rel="prefetch" href="/assets/index.html-D7Nph3ek.js" as="script"><link rel="prefetch" href="/assets/index.html-bO5aQHC3.js" as="script"><link rel="prefetch" href="/assets/index.html-oWXVgFcb.js" as="script"><link rel="prefetch" href="/assets/index.html-BtvxEcyn.js" as="script"><link rel="prefetch" href="/assets/index.html-SVI69W9E.js" as="script"><link rel="prefetch" href="/assets/index.html-BbxuFSWY.js" as="script"><link rel="prefetch" href="/assets/index.html-BFe9KGsX.js" as="script"><link rel="prefetch" href="/assets/index.html-CcUcGsU8.js" as="script"><link rel="prefetch" href="/assets/index.html-DuXvOCtn.js" as="script"><link rel="prefetch" href="/assets/index.html-CsYVP1Ee.js" as="script"><link rel="prefetch" href="/assets/index.html-BZNXvoee.js" as="script"><link rel="prefetch" href="/assets/index.html-DffBPlth.js" as="script"><link rel="prefetch" href="/assets/index.html-D5byNOA9.js" as="script"><link rel="prefetch" href="/assets/index.html-BIpJcaMq.js" as="script"><link rel="prefetch" href="/assets/index.html-Cclvrn0C.js" as="script"><link rel="prefetch" href="/assets/index.html-DBRMJnkP.js" as="script"><link rel="prefetch" href="/assets/index.html-C4wOU-pC.js" as="script"><link rel="prefetch" href="/assets/index.html-D46SKBHM.js" as="script"><link rel="prefetch" href="/assets/index.html-BJzbp6fj.js" as="script"><link rel="prefetch" href="/assets/index.html-BuuWC_5c.js" as="script"><link rel="prefetch" href="/assets/index.html-BdXimPXg.js" as="script"><link rel="prefetch" href="/assets/index.html-Co7EU3KZ.js" as="script"><link rel="prefetch" href="/assets/index.html-DLv0WxRB.js" as="script"><link rel="prefetch" href="/assets/index.html-T3K5oxvg.js" as="script"><link rel="prefetch" href="/assets/index.html-dRmPcSI7.js" as="script"><link rel="prefetch" href="/assets/index.html-B0QIdxhL.js" as="script"><link rel="prefetch" href="/assets/index.html-DR8uFVGu.js" as="script"><link rel="prefetch" href="/assets/index.html-BqRbC-7N.js" as="script"><link rel="prefetch" href="/assets/index.html-DnflFzVH.js" as="script"><link rel="prefetch" href="/assets/index.html-CoTzCY4j.js" as="script"><link rel="prefetch" href="/assets/index.html-Bg9CM87b.js" as="script"><link rel="prefetch" href="/assets/index.html-DG8mpR12.js" as="script"><link rel="prefetch" href="/assets/index.html-BWVnAnPc.js" as="script"><link rel="prefetch" href="/assets/index.html-dPFDGAnR.js" as="script"><link rel="prefetch" href="/assets/index.html-DKU3ACQJ.js" as="script"><link rel="prefetch" href="/assets/index.html-BgklZS89.js" as="script"><link rel="prefetch" href="/assets/index.html-DcsZqy80.js" as="script"><link rel="prefetch" href="/assets/index.html-BqSD2tD7.js" as="script"><link rel="prefetch" href="/assets/index.html-DruOQWAv.js" as="script"><link rel="prefetch" href="/assets/index.html-A9a2zzOf.js" as="script"><link rel="prefetch" href="/assets/index.html-D8r7fOa3.js" as="script"><link rel="prefetch" href="/assets/index.html-BXN1kcgZ.js" as="script"><link rel="prefetch" href="/assets/index.html-6s8u9tWA.js" as="script"><link rel="prefetch" href="/assets/index.html-C1tWJLKR.js" as="script"><link rel="prefetch" href="/assets/index.html-Ck8QB2Yk.js" as="script"><link rel="prefetch" href="/assets/index.html-vvppFMQ8.js" as="script"><link rel="prefetch" href="/assets/index.html-CoTDsdMc.js" as="script"><link rel="prefetch" href="/assets/index.html-DFWqG11e.js" as="script"><link rel="prefetch" href="/assets/index.html-DBQ_Pf_a.js" as="script"><link rel="prefetch" href="/assets/index.html-CogZr12k.js" as="script"><link rel="prefetch" href="/assets/index.html-CG6pkqpU.js" as="script"><link rel="prefetch" href="/assets/index.html-Qe4RZEFP.js" as="script"><link rel="prefetch" href="/assets/index.html-B36z9cJv.js" as="script"><link rel="prefetch" href="/assets/index.html-fTMvluKk.js" as="script"><link rel="prefetch" href="/assets/index.html-DCNBrLRQ.js" as="script"><link rel="prefetch" href="/assets/index.html-CGsNPNa0.js" as="script"><link rel="prefetch" href="/assets/index.html-BqVyC19r.js" as="script"><link rel="prefetch" href="/assets/index.html-D2nf00ow.js" as="script"><link rel="prefetch" href="/assets/index.html-COInHYqu.js" as="script"><link rel="prefetch" href="/assets/index.html-Chbm4mH6.js" as="script"><link rel="prefetch" href="/assets/index.html-W2B0dRwI.js" as="script"><link rel="prefetch" href="/assets/index.html-BUlkpK18.js" as="script"><link rel="prefetch" href="/assets/index.html-Cfs9L7fI.js" as="script"><link rel="prefetch" href="/assets/index.html-GVGxHVLE.js" as="script"><link rel="prefetch" href="/assets/index.html-Ch2SyZPw.js" as="script"><link rel="prefetch" href="/assets/404.html-BQ08fjuR.js" as="script"><link rel="prefetch" href="/assets/index.html-BjVEtomp.js" as="script"><link rel="prefetch" href="/assets/index.html-I7M9H6kr.js" as="script"><link rel="prefetch" href="/assets/index.html-C9hA5HtA.js" as="script"><link rel="prefetch" href="/assets/SearchBox-BquNHWNe.js" as="script"></head><body><div id="app"><!--[--><div class="theme-plume" data-v-0f4179d5><!--[--><!--[--><span tabindex="-1" data-v-ab0f771b></span><a href="#LayoutContent" class="skip-link visually-hidden" data-v-ab0f771b> Skip to content </a><!--]--><!----><div class="nav-wrapper" data-v-0f4179d5 data-v-4aa7bc5a><div class="navbar-wrapper has-sidebar" data-v-4aa7bc5a data-v-f397576e><div class="container" data-v-f397576e><div class="title" data-v-f397576e><div class="navbar-title has-sidebar" data-v-f397576e data-v-cdd38883><a class="auto-link link title" href="/" data-v-cdd38883 data-v-66dcefcc><!--[--><!--[--><!--[--><!--[--><img class="plume-image dark logo" src="/avatar.jpeg" alt data-v-00a6d3ab><!--]--><!--[--><img class="plume-image light logo" src="/avatar.jpeg" alt data-v-00a6d3ab><!--]--><!--]--><!--]--> <!--]--><!----></a></div></div><div class="content" data-v-f397576e><div class="curtain" data-v-f397576e></div><div class="content-body" data-v-f397576e><div class="navbar-search search" data-v-f397576e><div class="search-wrapper" data-v-f7169ce8><!----><div id="local-search" data-v-f7169ce8><button type="button" class="mini-search mini-search-button" aria-label="搜索文档" data-v-f7169ce8><span class="mini-search-button-container"><svg class="mini-search-search-icon" width="20" height="20" viewBox="0 0 20 20" aria-label="search icon"><path d="M14.386 14.386l4.0877 4.0877-4.0877-4.0877c-2.9418 2.9419-7.7115 2.9419-10.6533 0-2.9419-2.9418-2.9419-7.7115 0-10.6533 2.9418-2.9419 7.7115-2.9419 10.6533 0 2.9419 2.9418 2.9419 7.7115 0 10.6533z" stroke="currentColor" fill="none" fill-rule="evenodd" stroke-linecap="round" stroke-linejoin="round"></path></svg><span class="mini-search-button-placeholder">搜索文档</span></span><span class="mini-search-button-keys"><kbd class="mini-search-button-key"></kbd><kbd class="mini-search-button-key">K</kbd></span></button></div></div></div><nav aria-labelledby="main-nav-aria-label" class="navbar-menu menu" data-v-f397576e data-v-a3f1a60e><span id="main-nav-aria-label" class="visually-hidden" data-v-a3f1a60e>Main Navigation</span><!--[--><!--[--><a class="auto-link link navbar-menu-link" href="/" data-v-a3f1a60e data-v-681c4d02 data-v-66dcefcc><!--[--><!----><i data-v-681c4d02>首页</i><!--]--><!----></a><!--]--><!--[--><a class="auto-link link navbar-menu-link active" href="/NGS/filetype/" data-v-a3f1a60e data-v-681c4d02 data-v-66dcefcc><!--[--><!----><i data-v-681c4d02>NGS</i><!--]--><!----></a><!--]--><!--[--><a class="auto-link link navbar-menu-link" href="/R/" data-v-a3f1a60e data-v-681c4d02 data-v-66dcefcc><!--[--><!----><i data-v-681c4d02>R</i><!--]--><!----></a><!--]--><!--[--><a class="auto-link link navbar-menu-link" href="/py/chara_py/" data-v-a3f1a60e data-v-681c4d02 data-v-66dcefcc><!--[--><!----><i data-v-681c4d02>Python</i><!--]--><!----></a><!--]--><!--[--><div class="flyout-wrapper navbar-menu-group" data-v-a3f1a60e data-v-8ad12054><button type="button" class="button" aria-haspopup="true" aria-expanded="false" data-v-8ad12054><span class="text" data-v-8ad12054><!----><!----><span data-v-8ad12054>Analysis</span><span class="vpi-chevron-down text-icon" data-v-8ad12054></span></span></button><div class="menu" data-v-8ad12054><div class="menu-wrapper" data-v-8ad12054 data-v-a471995b><div class="items" data-v-a471995b><!--[--><!--[--><div class="menu-link" data-v-a471995b data-v-6e20746d><a class="auto-link link" href="/scRNA/" data-v-6e20746d data-v-66dcefcc><!--[--><!----><i data-v-6e20746d>scRNA</i><!--]--><!----></a></div><!--]--><!--]--></div><!--[--><!--]--></div></div></div><!--]--><!--[--><div class="flyout-wrapper navbar-menu-group" data-v-a3f1a60e data-v-8ad12054><button type="button" class="button" aria-haspopup="true" aria-expanded="false" data-v-8ad12054><span class="text" data-v-8ad12054><!----><!----><span data-v-8ad12054>资源</span><span class="vpi-chevron-down text-icon" data-v-8ad12054></span></span></button><div class="menu" data-v-8ad12054><div class="menu-wrapper" data-v-8ad12054 data-v-a471995b><div class="items" data-v-a471995b><!--[--><!--[--><div class="menu-link" data-v-a471995b data-v-6e20746d><a class="auto-link link" href="/bioinfo/" data-v-6e20746d data-v-66dcefcc><!--[--><!----><i data-v-6e20746d>学习</i><!--]--><!----></a></div><!--]--><!--]--></div><!--[--><!--]--></div></div></div><!--]--><!--[--><div class="flyout-wrapper navbar-menu-group" data-v-a3f1a60e data-v-8ad12054><button type="button" class="button" aria-haspopup="true" aria-expanded="false" data-v-8ad12054><span class="text" data-v-8ad12054><!----><!----><span data-v-8ad12054>Awesome</span><span class="vpi-chevron-down text-icon" data-v-8ad12054></span></span></button><div class="menu" data-v-8ad12054><div class="menu-wrapper" data-v-8ad12054 data-v-a471995b><div class="items" data-v-a471995b><!--[--><!--[--><div class="menu-link" data-v-a471995b data-v-6e20746d><a class="auto-link link" href="/awesome/command/" data-v-6e20746d data-v-66dcefcc><!--[--><!----><i data-v-6e20746d>Command</i><!--]--><!----></a></div><!--]--><!--[--><div class="menu-link" data-v-a471995b data-v-6e20746d><a class="auto-link link" href="/awesome/biotools/" data-v-6e20746d data-v-66dcefcc><!--[--><!----><i data-v-6e20746d>Tools</i><!--]--><!----></a></div><!--]--><!--[--><div class="menu-link" data-v-a471995b data-v-6e20746d><a class="auto-link link" href="/awesome/Rpkgs/" data-v-6e20746d data-v-66dcefcc><!--[--><!----><i data-v-6e20746d>R package</i><!--]--><!----></a></div><!--]--><!--[--><div class="menu-link" data-v-a471995b data-v-6e20746d><a class="auto-link link" href="/awesome/Plots/" data-v-6e20746d data-v-66dcefcc><!--[--><!----><i data-v-6e20746d>Plot</i><!--]--><!----></a></div><!--]--><!--]--></div><!--[--><!--]--></div></div></div><!--]--><!--[--><a class="auto-link link navbar-menu-link" href="/blog/" data-v-a3f1a60e data-v-681c4d02 data-v-66dcefcc><!--[--><!----><i data-v-681c4d02>博客</i><!--]--><!----></a><!--]--><!--]--></nav><!----><div class="navbar-appearance appearance" data-v-f397576e data-v-2e93bdf2><button class="switch-wrapper switch-appearance" type="button" role="switch" aria-label="toggle dark mode" aria-checked="false" data-v-2e93bdf2 data-v-61e5dfef data-v-492a9bf8><span class="check" data-v-492a9bf8><span class="icon" data-v-492a9bf8><!--[--><span class="vpi-sun sun" data-v-61e5dfef></span><span class="vpi-moon moon" data-v-61e5dfef></span><!--]--></span></span></button></div><div class="social-links navbar-social-links social-links" data-v-f397576e data-v-e47592f0 data-v-82034cd4><!--[--><a class="social-link" href="https://github.com/Moonerss" aria-label="github" target="_blank" rel="noopener" data-v-82034cd4 data-v-595ca7ec><span class="vpi-social-github" /></a><!--]--></div><div class="flyout-wrapper navbar-extra extra" data-v-f397576e data-v-de8fb2b4 data-v-8ad12054><button type="button" class="button" aria-haspopup="true" aria-expanded="false" aria-label="extra navigation" data-v-8ad12054><span class="vpi-more-horizontal icon" data-v-8ad12054></span></button><div class="menu" data-v-8ad12054><div class="menu-wrapper" data-v-8ad12054 data-v-a471995b><!----><!--[--><!--[--><!----><div class="group" data-v-de8fb2b4><div class="item appearance" data-v-de8fb2b4><p class="label" data-v-de8fb2b4>外观</p><div class="appearance-action" data-v-de8fb2b4><button class="switch-wrapper switch-appearance" type="button" role="switch" aria-label="toggle dark mode" aria-checked="false" data-v-de8fb2b4 data-v-61e5dfef data-v-492a9bf8><span class="check" data-v-492a9bf8><span class="icon" data-v-492a9bf8><!--[--><span class="vpi-sun sun" data-v-61e5dfef></span><span class="vpi-moon moon" data-v-61e5dfef></span><!--]--></span></span></button></div></div></div><div class="group" data-v-de8fb2b4><div class="item social-links" data-v-de8fb2b4><div class="social-links social-links-list" data-v-de8fb2b4 data-v-82034cd4><!--[--><a class="social-link" href="https://github.com/Moonerss" aria-label="github" target="_blank" rel="noopener" data-v-82034cd4 data-v-595ca7ec><span class="vpi-social-github" /></a><!--]--></div></div></div><!--]--><!--]--></div></div></div><button type="button" class="navbar-hamburger hamburger" aria-label="mobile navigation" aria-expanded="false" aria-controls="nav-screen" data-v-f397576e data-v-1925941b><span class="container" data-v-1925941b><span class="top" data-v-1925941b></span><span class="middle" data-v-1925941b></span><span class="bottom" data-v-1925941b></span></span></button></div></div></div></div><!----></div><div class="local-nav reached-top" data-v-0f4179d5 data-v-0b456a43><button class="menu" aria-expanded="false" aria-controls="SidebarNav" data-v-0b456a43><span class="vpi-align-left menu-icon" data-v-0b456a43></span><span class="menu-text" data-v-0b456a43>Menu</span></button><div class="local-nav-outline-dropdown" style="--vp-vh:0px;" data-v-0b456a43 data-v-f6c5b43d><button class="" data-v-f6c5b43d>此页内容 <span class="vpi-chevron-right icon" data-v-f6c5b43d></span></button><!----></div></div><aside class="sidebar-wrapper" data-v-0f4179d5 data-v-e5d2823d><div class="curtain" data-v-e5d2823d></div><nav id="SidebarNav" class="nav" aria-labelledby="sidebar-aria-label" tabindex="-1" data-v-e5d2823d><span id="sidebar-aria-label" class="visually-hidden" data-v-e5d2823d> Sidebar Navigation </span><!--[--><div class="group" data-v-e5d2823d><section class="sidebar-item level-0 collapsible has-active" data-v-e5d2823d data-v-19611753><div class="item" role="button" tabindex="0" data-v-19611753><div class="indicator" data-v-19611753></div><!----><h2 class="text" data-v-19611753>基础内容</h2><div class="caret" role="button" aria-label="toggle section" tabindex="0" data-v-19611753><span class="vpi-chevron-right caret-icon" data-v-19611753></span></div></div><div class="items" data-v-19611753><!--[--><div class="sidebar-item level-1 is-link" data-v-19611753 data-v-19611753><div class="item" tabindex="0" data-v-19611753><div class="indicator" data-v-19611753></div><!----><a class="auto-link link link" href="/NGS/filetype/" data-v-19611753 data-v-66dcefcc><!--[--><p class="text" data-v-19611753>测序常见文件格式</p><!--]--><!----></a><!----></div><!----></div><div class="sidebar-item level-1 is-link" data-v-19611753 data-v-19611753><div class="item" tabindex="0" data-v-19611753><div class="indicator" data-v-19611753></div><!----><a class="auto-link link link" href="/NGS/concept/" data-v-19611753 data-v-66dcefcc><!--[--><p class="text" data-v-19611753>基础概念</p><!--]--><!----></a><!----></div><!----></div><div class="sidebar-item level-1 is-link" data-v-19611753 data-v-19611753><div class="item" tabindex="0" data-v-19611753><div class="indicator" data-v-19611753></div><!----><a class="auto-link link link" href="/NGS/genome/" data-v-19611753 data-v-66dcefcc><!--[--><p class="text" data-v-19611753>参考基因组知识汇总</p><!--]--><!----></a><!----></div><!----></div><div class="sidebar-item level-1 is-link" data-v-19611753 data-v-19611753><div class="item" tabindex="0" data-v-19611753><div class="indicator" data-v-19611753></div><!----><a class="auto-link link link" href="/NGS/gene_id/" data-v-19611753 data-v-66dcefcc><!--[--><p class="text" data-v-19611753>基因ID转换</p><!--]--><!----></a><!----></div><!----></div><div class="sidebar-item level-1 is-link" data-v-19611753 data-v-19611753><div class="item" tabindex="0" data-v-19611753><div class="indicator" data-v-19611753></div><!----><a class="auto-link link link" href="/NGS/tpm_fpkm_rpkm/" data-v-19611753 data-v-66dcefcc><!--[--><p class="text" data-v-19611753>TPM、RPKM与FPKM</p><!--]--><!----></a><!----></div><!----></div><div class="sidebar-item level-1 is-link" data-v-19611753 data-v-19611753><div class="item" tabindex="0" data-v-19611753><div class="indicator" data-v-19611753></div><!----><a class="auto-link link link" href="/NGS/vcf/" data-v-19611753 data-v-66dcefcc><!--[--><p class="text" data-v-19611753>VAF、MAF、MCF、CCF</p><!--]--><!----></a><!----></div><!----></div><div class="sidebar-item level-1 is-link" data-v-19611753 data-v-19611753><div class="item" tabindex="0" data-v-19611753><div class="indicator" data-v-19611753></div><!----><a class="auto-link link link" href="/NGS/epigenetics/" data-v-19611753 data-v-66dcefcc><!--[--><p class="text" data-v-19611753>表观遗传学的三个方向与测序方法</p><!--]--><!----></a><!----></div><!----></div><!--]--></div></section></div><!--]--></nav></aside><div id="LayoutContent" class="layout-content has-sidebar" data-v-0f4179d5 data-v-bdfa9fd7><!--[--><div class="has-sidebar has-aside plume-page" data-v-5c470ced><div class="container" data-v-5c470ced><div class="aside" data-v-5c470ced><div class="aside-container" data-v-5c470ced><div class="aside-content" data-v-5c470ced><div class="page-aside" data-v-5c470ced data-v-f97b8b42><div class="has-outline page-aside-outline" data-v-f97b8b42><div class="content" data-v-f97b8b42><div class="outline-marker" data-v-f97b8b42></div><div class="outline-title" data-v-f97b8b42><span data-v-f97b8b42>此页内容</span><span class="vpi-print icon" data-v-f97b8b42></span></div><nav aria-labelledby="doc-outline-aria-label" data-v-f97b8b42><span id="doc-outline-aria-label" class="visually-hidden" data-v-f97b8b42> Table of Contents for current page </span><ul class="root" data-v-f97b8b42 data-v-784c5b47><!--[--><li data-v-784c5b47><a class="outline-link" href="#fastq" data-v-784c5b47>FASTQ</a><ul class="nested" data-v-784c5b47 data-v-784c5b47><!--[--><li data-v-784c5b47><a class="outline-link" href="#质量编码格式" data-v-784c5b47>质量编码格式</a><!----></li><li data-v-784c5b47><a class="outline-link" href="#如何判断是phred33还是phred64" data-v-784c5b47>如何判断是Phred33还是Phred64</a><!----></li><!--]--></ul></li><li data-v-784c5b47><a class="outline-link" href="#fasta" data-v-784c5b47>FASTA</a><!----></li><li data-v-784c5b47><a class="outline-link" href="#sam" data-v-784c5b47>SAM</a><!----></li><li data-v-784c5b47><a class="outline-link" href="#bam" data-v-784c5b47>BAM</a><!----></li><li data-v-784c5b47><a class="outline-link" href="#vcf" data-v-784c5b47>VCF</a><!----></li><li data-v-784c5b47><a class="outline-link" href="#gtf和gff" data-v-784c5b47>GTF和GFF</a><!----></li><li data-v-784c5b47><a class="outline-link" href="#genepred" data-v-784c5b47>GenePred</a><!----></li><li data-v-784c5b47><a class="outline-link" href="#bed" data-v-784c5b47>BED</a><!----></li><li data-v-784c5b47><a class="outline-link" href="#maf" data-v-784c5b47>MAF</a><!----></li><li data-v-784c5b47><a class="outline-link" href="#wiggle、bigwig和bedgraph" data-v-784c5b47>Wiggle、BigWig和bedgraph</a><!----></li><!--]--></ul></nav></div></div></div></div></div></div><div class="content" data-v-5c470ced><div class="content-container" data-v-5c470ced><!--[--><!----><h1 class="page-title" data-v-d86f3e1b>测序常见文件格式</h1><div class="page-meta-wrapper" data-v-d86f3e1b><p class="author" data-v-d86f3e1b><span class="icon vpi-user" data-v-d86f3e1b></span><span data-v-d86f3e1b>Jeason</span></p><p class="reading-time" data-v-d86f3e1b><span class="vpi-books icon" data-v-d86f3e1b></span><span data-v-d86f3e1b>12207字</span><span data-v-d86f3e1b>约41分钟</span></p><!----><p class="create-time" data-v-d86f3e1b><span class="vpi-clock icon" data-v-d86f3e1b></span><span data-v-d86f3e1b>2024-05-10</span></p></div><!--]--><!--[--><div style="position:relative;" data-v-5c470ced><div class="plume-content" data-v-5c470ced><h1 id="生物信息学常见文件格式" tabindex="-1"><a class="header-anchor" href="#生物信息学常见文件格式"><span>生物信息学常见文件格式</span></a></h1><h2 id="fastq" tabindex="-1"><a class="header-anchor" href="#fastq"><span>FASTQ</span></a></h2><p>FASTQ用于保存生物序列（通常是核酸序列）和其测序质量信息的标准格式。 其序列以及质量信息都是使用一个ASCII字符标示，最初由Sanger开发。 目的是将FASTA序列与质量数据放到一起，目前已经成为高通量测序结果的实施标准。</p><p><strong>FASTQ文件中每个序列通常有四行</strong>：</p><ul><li>第一行是序列标识以及相关的描述信息，以‘@’开头</li><li>第二行是序列</li><li>第三行以‘+’开头，后面是序列标示符、描述信息，或者什么也不加，但是“+”不能少。</li><li>第四行，是质量信息，和第二行的序列相对应，每一个序列都有一个质量评分，根据评分体系的不同，每个字符的含义表示的数字也不相同。</li></ul><p>具体示例如下：</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>@SEQ_ID</span></span>
<span class="line"><span>GATTTGGGGTTCAAAGCAGTATCGATCAAATAGTAAATCCATTTGTTCAACTCACAGTTT</span></span>
<span class="line"><span>+</span></span>
<span class="line"><span>!&#39;&#39;*((((***+))%%%++)(%%%%).1***-+*&#39;&#39;))**55CCF&gt;&gt;&gt;&gt;&gt;&gt;CCCCCCC65</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p>上面说到第一行是序列标识以及相关的描述信息，以‘@’开头。可以像上面的示例那么简单，但如果是正规测序仪下机的真实数据，通常会很复杂。比如：</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>@EAS139:136:FC706VJ:2:2104:15343:197393 1:Y:18:ATCACG</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div></div></div><p>这个序列标识以及相关描述信息以 <code>:</code>分割，每一个字段信息如下：</p><table><thead><tr><th>字段</th><th>解释</th></tr></thead><tbody><tr><td>EAS139</td><td>the unique instrument name</td></tr><tr><td>136</td><td>the run id</td></tr><tr><td>FC706VJ</td><td>the flowcell id</td></tr><tr><td>2</td><td>flowcell lane</td></tr><tr><td>2104</td><td>tile number within the flowcell lane</td></tr><tr><td>15343</td><td>‘x’-coordinate of the cluster within the tile</td></tr><tr><td>197393</td><td>‘y’-coordinate of the cluster within the tile</td></tr><tr><td>1</td><td>the member of a pair, 1 or 2 (paired-end or mate-pair reads only)</td></tr><tr><td>Y</td><td>Y if the read fails filter (read is bad), N otherwise</td></tr><tr><td>18</td><td>0 when none of the control bits are on, otherwise it is an even number</td></tr><tr><td>ATCACG</td><td>index sequence</td></tr></tbody></table><p>当然，上面的表格介绍的只是其中一个测序仪下机数据，如果是其它机器，产商可以自由定义标识符格式，因为fastq格式的第一行只需要以@符号开头即可。</p><p>不过，也有一些时候fastq数据并不是测序仪直接下机的，而且他人上传到了NCBI的SRA中心，我们下载下来解压后一般就没有了测序仪相关的标识，例子如下：</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>@SRR001666.1 071112_SLXA-EAS1_s_7:5:1:817:345 length=36</span></span>
<span class="line"><span>GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACC</span></span>
<span class="line"><span>+SRR001666.1 071112_SLXA-EAS1_s_7:5:1:817:345 length=36</span></span>
<span class="line"><span>IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9IC</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><h3 id="质量编码格式" tabindex="-1"><a class="header-anchor" href="#质量编码格式"><span>质量编码格式</span></a></h3><p>质量评分指的是一个碱基的错误概率的对数值。 其最初在Phred拼接软件中定义与使用，其后在许多软件中得到使用。 其质量得分与错误概率的对应关系见下表：</p><table><thead><tr><th>PHRED QUALITY SCORE</th><th>PROBABILITY OF INCORRECT BASE CALL</th><th>BASE CALL ACCURACY</th></tr></thead><tbody><tr><td>10</td><td>1 in 10</td><td>90 %</td></tr><tr><td>20</td><td>1 in 100</td><td>99%</td></tr><tr><td>30</td><td>1 in 1000</td><td>99.9%</td></tr><tr><td>40</td><td>1 in 10000</td><td>99.99%</td></tr><tr><td>50</td><td>1 in 100000</td><td>99.999%</td></tr></tbody></table><blockquote><p>Phred quality scores Q are defined as a property which is logarithmically related to the base-calling error probabilities P. Q=-10lgP</p></blockquote><p>除了Phred质量得分换算标准，还有就是Solexa标准：是把P换成p/(1-p)</p><p>对于每个碱基的质量编码标示，不同的软件采用不同的方案，目前有5种方案：</p><ol><li>Sanger，Phred quality score，值的范围从0到92，对应的ASCII码从33到126，但是对于测序数据（raw read data）质量得分通常小于60，序列拼接或者mapping可能用到更大的分数。</li><li>Solexa/Illumina 1.0, Solexa/Illumina quality score，值的范围从-5到63，对应的ASCII码从59到126，对于测序数据，得分一般在-5到40之间。</li><li>Illumina 1.3+，Phred quality score，值的范围从0到62对应的ASCII码从64到126，低于测序数据，得分在0到40之间；</li><li>Illumina 1.5+，Phred quality score，但是0到2作为另外的标示。</li><li>Illumina 1.8+</li></ol><p>下面是更为直观的表示：</p><p><img src="https://cdn.jsdelivr.net/gh/Moonerss/CDN/paper/filtype/wiki-fastq-format.png" alt=""></p><h3 id="如何判断是phred33还是phred64" tabindex="-1"><a class="header-anchor" href="#如何判断是phred33还是phred64"><span>如何判断是Phred33还是Phred64</span></a></h3><p>默认读取1000条序列，在这1000条序列中：</p><ul><li>如果有2个以上的质量字符ASCII值小于等于58（即有两个碱基的得分小于等于25），同时没有任何质量字符的ASCII值大于等于75，即判断是Phred+33。</li><li>如果有2个以上的质量字符ASCII值大于等于75（即有两个碱基的得分大于等于10），同时没有任何质量字符的ASCII值小于等于58，即判断是Phred+64。</li><li>如果所有质量字符的ASCII值介于59到74之间，即判断可能是Phred+33，但建议使用更多的序列做进一步测试。出现这种结果可能有两种情况：(1) Phred+33编码，所有碱基质量得分介于26到42之间；(2)Phred+64编码，所有碱基质量得分介于-5到10。是前者的可能性大。</li><li>如果出现上述3种以外的情况，建议打印出质量字符的ASCII值人工判断，脚本如下：</li></ul><div class="language-bash line-numbers-mode" data-ext="sh" data-title="sh"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span style="--shiki-light:#59873A;--shiki-dark:#80A665;">cat</span><span style="--shiki-light:#B56959;--shiki-dark:#C98A7D;"> test.fq</span><span style="--shiki-light:#AB5959;--shiki-dark:#CB7676;"> |</span><span style="--shiki-light:#59873A;--shiki-dark:#80A665;"> head</span><span style="--shiki-light:#A65E2B;--shiki-dark:#C99076;"> -n</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;"> 1000</span><span style="--shiki-light:#AB5959;--shiki-dark:#CB7676;"> |</span><span style="--shiki-light:#59873A;--shiki-dark:#80A665;"> awk</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;"> &#39;</span><span style="--shiki-light:#B56959;--shiki-dark:#C98A7D;">{if(NR%4==0) printf(&quot;%s&quot;,$0);}</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&#39;</span><span style="--shiki-light:#A65E2B;--shiki-dark:#C99076;"> \</span></span>
<span class="line"><span style="--shiki-light:#AB5959;--shiki-dark:#CB7676;">|</span><span style="--shiki-light:#59873A;--shiki-dark:#80A665;"> od</span><span style="--shiki-light:#A65E2B;--shiki-dark:#C99076;"> -A</span><span style="--shiki-light:#B56959;--shiki-dark:#C98A7D;"> n</span><span style="--shiki-light:#A65E2B;--shiki-dark:#C99076;"> -t</span><span style="--shiki-light:#B56959;--shiki-dark:#C98A7D;"> u1</span><span style="--shiki-light:#A65E2B;--shiki-dark:#C99076;"> -v</span><span style="--shiki-light:#A65E2B;--shiki-dark:#C99076;"> \</span></span>
<span class="line"><span style="--shiki-light:#AB5959;--shiki-dark:#CB7676;">|</span><span style="--shiki-light:#59873A;--shiki-dark:#80A665;"> awk</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;"> &#39;</span><span style="--shiki-light:#B56959;--shiki-dark:#C98A7D;">BEGIN{min=100;max=0;} \</span></span>
<span class="line"><span style="--shiki-light:#B56959;--shiki-dark:#C98A7D;">{for(i=1;i&lt;=NF;i++) {if($i&gt;max) max=$i; if($i&lt;min) min=$i;}}END \</span></span>
<span class="line"><span style="--shiki-light:#B56959;--shiki-dark:#C98A7D;">{if(max&lt;=126 &amp;&amp; min&lt;59) print &quot;Phred33&quot;; \</span></span>
<span class="line"><span style="--shiki-light:#B56959;--shiki-dark:#C98A7D;">else if(max&gt;73 &amp;&amp; min&gt;=64) print &quot;Phred64&quot;; \</span></span>
<span class="line"><span style="--shiki-light:#B56959;--shiki-dark:#C98A7D;">else if(min&gt;=59 &amp;&amp; min&lt;64 &amp;&amp; max&gt;73) print &quot;Solexa64&quot;; \</span></span>
<span class="line"><span style="--shiki-light:#B56959;--shiki-dark:#C98A7D;">else print &quot;Unknown score encoding&quot;; \</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><h2 id="fasta" tabindex="-1"><a class="header-anchor" href="#fasta"><span>FASTA</span></a></h2><p>FASTA格式用于表示核苷酸序列或氨基酸序列的格式。 在这种格式中碱基对或氨基酸用单个字母来编码，且允许在序列前添加序列名及注释。 fasta序列格式是blast组织数据的基本格式，无论是数据库还是查询序列，大多数情况都使用fasta序列格式。</p><p><strong>FASTA格式文件一般包含两行</strong>:</p><ul><li>第一行: 以 <code>&gt;</code>标识符开始，包含序列描述信息</li><li>第二行: 基因或蛋白序列</li></ul><p>下面是一个来源于NCBI的fasta格式序列：</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>&gt;gi|187608668|ref|NM_001043364.2| Bombyx mori moricin (Mor), mRNA</span></span>
<span class="line"><span>AAACCGCGCAGTTATTTAAAATATGAATATTTTAAAACTTTTCTTTGTTTTTA</span></span>
<span class="line"><span>TTGTGGCAATGTCTCTGGTGTCATGTAGTACAGCCGCTCCAGCAAAAATACCT</span></span>
<span class="line"><span>ATCAAGGCCATTAAGACTGTAGGAAAGGCAGTCGGTAAAGGTCTAAGAGCCAT </span></span>
<span class="line"><span>ATCAAGGCCATTAA</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p>Fasta格式首先以大于号 <code>&gt;</code>开头，接着是序列的标识符“gi|187608668|ref|NM_001043364.2|”，然后是序列的描述信息。 换行后是序列信息，序列中允许空格，换行，空行，直到下一个大于号，表示该序列的结束。 下面简单给一个表格说明序列来源的数据库与对应的标识符</p><table><thead><tr><th>Database Name数据库名称</th><th>Identifier Syntax 标识符</th></tr></thead><tbody><tr><td>GenBank</td><td>gb</td></tr><tr><td>EMBL Data Library</td><td>emb</td></tr><tr><td>DDBJ, DNA Database of Japan</td><td>dbj</td></tr><tr><td>NBRF PIR</td><td>pir</td></tr><tr><td>Protein Research Foundation</td><td>prf</td></tr><tr><td>SWISS-PROT</td><td>sp</td></tr><tr><td>Brookhaven Protein Data Bank</td><td>pdb</td></tr><tr><td>Patents</td><td>pat</td></tr><tr><td>GenInfo Backbone Id</td><td>bbs</td></tr><tr><td>General database identifier</td><td>gnl</td></tr><tr><td>NCBI Reference Sequence</td><td>ref</td></tr><tr><td>Local Sequence identifier</td><td>lcl</td></tr></tbody></table><p>通常情况下序列的标识符不会像上面的例子那样复杂，再复杂的标识符也是有规则的，上面的标识符是NCBI定义的，可以去其官网了解详情。</p><p><strong>序列中字母代表的含义</strong></p><p>FASTA格式支持的核苷酸代码如下：</p><table><thead><tr><th>核苷酸代码</th><th>意义</th></tr></thead><tbody><tr><td>A</td><td>Adenosine</td></tr><tr><td>C</td><td>Cytosine</td></tr><tr><td>G</td><td>Guanine</td></tr><tr><td>T</td><td>Thymidine</td></tr><tr><td>U</td><td>Uracil</td></tr><tr><td>R</td><td>G A (puRine)</td></tr><tr><td>Y</td><td>T C (pYrimidine)</td></tr><tr><td>K</td><td>G T (Ketone)</td></tr><tr><td>M</td><td>A C (aMino group)</td></tr><tr><td>S</td><td>G C (Strong interaction)</td></tr><tr><td>W</td><td>A T (Weak interaction)</td></tr><tr><td>B</td><td>G T C (not A) (B comes after A)</td></tr><tr><td>D</td><td>G A T (not C) (D comes after C)</td></tr><tr><td>H</td><td>A C T (not G) (H comes after G)</td></tr><tr><td>V</td><td>G C A (not T, not U) (V comes after U)</td></tr><tr><td>N</td><td>A G C T (aNy)</td></tr><tr><td>X</td><td>masked</td></tr><tr><td>-</td><td>gap of indeterminate length</td></tr></tbody></table><p>FASTA格式支持的氨基酸代码如下：</p><table><thead><tr><th>氨基酸代码</th><th>意义</th></tr></thead><tbody><tr><td>A</td><td>Alanine</td></tr><tr><td>B</td><td>Aspartic acid or Asparagine</td></tr><tr><td>C</td><td>Cysteine</td></tr><tr><td>D</td><td>Aspartic acid</td></tr><tr><td>E</td><td>Glutamic acid</td></tr><tr><td>F</td><td>Phenylalanine</td></tr><tr><td>G</td><td>Glycine</td></tr><tr><td>H</td><td>Histidine</td></tr><tr><td>I</td><td>Isoleucine</td></tr><tr><td>K</td><td>Lysine</td></tr><tr><td>L</td><td>Leucine</td></tr><tr><td>M</td><td>Methionine</td></tr><tr><td>N</td><td>Asparagine</td></tr><tr><td>O</td><td>Pyrrolysine</td></tr><tr><td>P</td><td>Proline</td></tr><tr><td>Q</td><td>Glutamine</td></tr><tr><td>R</td><td>Arginine</td></tr><tr><td>S</td><td>Serine</td></tr><tr><td>T</td><td>Threonine</td></tr><tr><td>U</td><td>Selenocysteine</td></tr><tr><td>V</td><td>Valine</td></tr><tr><td>W</td><td>Tryptophan</td></tr><tr><td>Y</td><td>Tyrosine</td></tr><tr><td>Z</td><td>Glutamic acid or Glutamine</td></tr><tr><td>X</td><td>any</td></tr><tr><td>*</td><td>translation stop</td></tr><tr><td>-</td><td>gap of indeterminate length</td></tr></tbody></table><p><strong>注意事项：</strong></p><p>对于自己构建的序列数据库（序列不是来源与NCBI或其他数据），可以采用“gnl|database|identifier”或者“lcl|identifier”格式，以保证可以使用blast的所有功能。 database或者identifier是需要指定的数据库的标识和序列标识，指定的名称可以用大小写字母、数字、下划线“_”、破折号“-”或者点号“.”。 注意名称是区分大小写的，同时不能出现空格，空格表示序列标识符结束。</p><p>数据库中的序列标识符必须保证唯一，许多时候格式数据库是formatdb报告错误，就是因为标示符重复，还有一点需要强调的是序列不能为空，否则也会报错。</p><h2 id="sam" tabindex="-1"><a class="header-anchor" href="#sam"><span>SAM</span></a></h2><p>SAM格式主要应用于测序序列mapping到基因组上的结果表示，当然也可以表示任意的多重比对结果。 SAM是一种序列比对格式标准，由sanger制定，是以TAB为分割符的文本格式。 SAM的全称是 <code>sequence alignment/map format</code>。</p><p>SAM分为两部分，注释信息 <code>header section</code>和比对结果部分 <code>alignment section</code>。 通常是把FASTQ文件格式的测序数据比对到对应的参考基因组版本得到的。 注释信息并不是SAM文件的重点，是该SAM文件产生以及被处理过程的一个记录，规定以 <code>@</code>开头，用不同的tag表示不同的信息，主要有：</p><ul><li><code>@HD</code>，说明符合标准的版本、对比序列的排列顺序；</li><li><code>@SQ</code>，参考序列说明；</li><li><code>@RG</code>，比对上的序列（read）说明；</li><li><code>@PG</code>，使用的程序说明；</li><li><code>@CO</code>，任意的说明信息。</li></ul><p>一个简单的SAM文件例子如下：</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>@HD VN:1.0 SO:unsorted</span></span>
<span class="line"><span>@SQ SN:chr1 LN:249250621</span></span>
<span class="line"><span>@SQ SN:chr2 LN:243199373</span></span>
<span class="line"><span>@PG ID:Bowtie VN:1.0.0 CL:&quot;bowtie genome/hg19 -q reads/SRR3101251.fastq -m 1 -p 4 -S&quot;</span></span>
<span class="line"><span>SRR3101251.1 0 chr19 9486878 255 49M * 0 0 NTACTCCCACTACTCTCAGATTCAAGCAATCCTCCCACCCTAGCCCACC #1=DDDFFHHHHHIHHIJJJHIJIIJIHIFHJIIJJJJJJJIIJJJJJJ XA:i:1 MD:Z:0A48 NM:i:1</span></span>
<span class="line"><span>SRR3101251.5 16 chr2 240279787 255 49M * 0 0 CCTGAATCCATCAGAGCAGCCGGGCTGTGACACTCACTGTCATGATGTT JIJJIHIIIIJJJJJJJJJGHJJJJIIHJHICJIGCHHHHHFFFFFCCC XA:i:0 MD:Z:49 NM:i:0</span></span>
<span class="line"><span>SRR3101251.6 4 * 0 0 * * 0 0 NATTCCCACCTATGAGTGAGAATATGCGGTGTTTGGTTTTTTGTTCTTG #1=DDDFFHHHHHJJJGHIJJJJJJJJJJCGGIIJJIIJJJIJHJIIJJ XM:i:1</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p><strong>比对结果详解</strong></p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>SRR035022.2621862 163 16 59999 37 22S54M = 60102 179 CCAACCCAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCGACCCTCACCCTCACCC &gt;AAA=&gt;?AA&gt;@@B@B?AABAB?AABAB?AAC@B?@AB@A?A&gt;A@A?AAAAB??ABAB?79A?AAB;B?@?@&lt;=8:8 XT:A:M XN:i:2 SM:i:37 AM:i:37 XM:i:0 XO:i:0 XG:i:0 RG:Z:SRR035022 NM:i:2 MD:Z:0N0N52 OQ:Z:CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCBCCCCCCBBCC@CCCCCCCCCCACCCCC;CCCBBC?CCCACCACA@</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div></div></div><p>每一个字段的说明如下：</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>QNAME	SRR035022.2621862</span></span>
<span class="line"><span>FLAG	163</span></span>
<span class="line"><span>RNAME	16</span></span>
<span class="line"><span>POS	59999</span></span>
<span class="line"><span>MAQ	37</span></span>
<span class="line"><span>CIGAR	22S54M</span></span>
<span class="line"><span>MRNM	=</span></span>
<span class="line"><span>MPOS	60102</span></span>
<span class="line"><span>ISIZE	179</span></span>
<span class="line"><span>SEQ	CCAACCCAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCGACCCTCACCCTCACCC</span></span>
<span class="line"><span>QUAL	&gt;AAA=&gt;?AA&gt;@@B@B?AABAB?AABAB?AAC@B?@AB@A?A&gt;A@A?AAAAB??ABAB?79A?AAB;B?@?@&lt;=8:8</span></span>
<span class="line"><span>TAG	XT:A:M</span></span>
<span class="line"><span>TAG	XN:i:2</span></span>
<span class="line"><span>TAG	SM:i:37</span></span>
<span class="line"><span>TAG	AM:i:37</span></span>
<span class="line"><span>TAG	XM:i:0</span></span>
<span class="line"><span>TAG	XO:i:0</span></span>
<span class="line"><span>TAG	XG:i:0</span></span>
<span class="line"><span>TAG	RG:Z:SRR035022</span></span>
<span class="line"><span>TAG	NM:i:2</span></span>
<span class="line"><span>TAG	MD:Z:0N0N52</span></span>
<span class="line"><span>TAG	OQ:Z:CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCBCCCCCCBBCC@CCCCCCCCCCACCCCC;CCCBBC?CCCACCACA</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p>比对结果部分（alignment section），每一行表示一个片段（segment）的比对信息，包括11个必须的字段（mandatory fields）和一个可选的字段，字段之间用tag分割。</p><p>必须的字段有11个，顺序固定，不可自行改动，根据字段定义，可以为 <code>0</code>或者 <code>*</code>，这是11个字段包括：</p><ol><li><code>QNAME</code>，比对片段的（template）的编号；</li><li><code>FLAG</code>，位标识，template mapping情况的数字表示，每一个数字代表一种比对情况，这里的值是符合情况的数字相加总和； （picard专门有一个工具解读sam的flag:http://broadinstitute.github.io/picard/explain-flags.html）</li></ol><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>1	The read is one of a pair  read是pair中的一条（read表示本条read，mate表示pair中的另一条read）</span></span>
<span class="line"><span>2	The alignment is one end of a proper paired-end alignment  pair一正一负完美的比对上</span></span>
<span class="line"><span>4	The read has no reported alignments  这条read没有比对上</span></span>
<span class="line"><span>8	The read is one of a pair and has no reported alignments  mate没有比对上</span></span>
<span class="line"><span>16	The alignment is to the reverse reference strand  这条read反向比对</span></span>
<span class="line"><span>32	The other mate in the paired-end alignment is aligned to the reverse reference strand   mate反向比对</span></span>
<span class="line"><span>64	The read is the first (#1) mate in a pair  这条read是read1</span></span>
<span class="line"><span>128	The read is the second (#2) mate in a pair  这条read是read2</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><ol start="3"><li><code>RNAME</code>，参考序列的编号，如果注释中对SQ-SN进行了定义，这里必须和其保持一致，另外对于没有mapping上的序列，这里是’*‘；</li><li><code>POS</code>，比对上的位置，注意是从1开始计数，没有比对上，此处为0；</li><li><code>MAPQ</code>，mappint的质量；</li><li><code>CIGAR</code>，简要比对信息表达式（Compact Idiosyncratic Gapped Alignment Report），其以参考序列为基础，使用数字加字母表示比对结果，比如3S6M1P1I4M，前三个碱基被剪切去除了，然后6个比对上了，然后打开了一个缺口，有一个碱基插入，最后是4个比对上了，是按照顺序的；</li></ol><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>“M”表示 match或 mismatch；</span></span>
<span class="line"><span>“I”表示 insert；</span></span>
<span class="line"><span>“D”表示 deletion；</span></span>
<span class="line"><span>“N”表示 skipped（跳过这段区域）；</span></span>
<span class="line"><span>“S”表示 soft clipping（被剪切的序列存在于序列中）；</span></span>
<span class="line"><span>“H”表示 hard clipping（被剪切的序列不存在于序列中）；</span></span>
<span class="line"><span>“P”表示 padding；</span></span>
<span class="line"><span>“=”表示 match；</span></span>
<span class="line"><span>“X”表示 mismatch（错配，位置是一一对应的）；</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><ol start="7"><li><code>RNEXT</code>，下一个片段比对上的参考序列的编号，没有另外的片段，这里是’*‘，同一个片段，用’=‘；</li><li><code>PNEXT</code>，下一个片段比对上的位置，如果不可用，此处为0；</li><li><code>TLEN</code>，Template的长度，最左边得为正，最右边的为负，中间的不用定义正负，不分区段（single-segment)的比对上，或者不可用时，此处为0；</li><li><code>SEQ</code>，序列片段的序列信息，如果不存储此类信息，此处为’*‘，注意CIGAR中M/I/S/=/X对应数字的和要等于序列长度；</li><li><code>QUAL</code>，序列的质量信息，格式同FASTQ一样。read质量的ASCII编码。</li><li>可选字段（optional fields)，格式如：TAG:TYPE:VALUE，其中TAG有两个大写字母组成，每个TAG代表一类信息，每一行一个TAG只能出现一次，TYPE表示TAG对应值的类型，可以是字符串、整数、字节、数组等。</li></ol><h2 id="bam" tabindex="-1"><a class="header-anchor" href="#bam"><span>BAM</span></a></h2><p>BAM本质上就是二进制压缩的SAM文件，大部分生物信息学流程都需要这个格式，为了节省存储空间以及方便索引。</p><div class="language-r line-numbers-mode" data-ext="r" data-title="r"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span style="--shiki-light:#A0ADA0;--shiki-dark:#758575DD;"># BiocInstaller::biocLite(&#39;Rsamtools&#39;)</span></span>
<span class="line"><span style="--shiki-light:#998418;--shiki-dark:#B8A965;">library</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">(</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">Rsamtools</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">) </span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">test_bam_file</span><span style="--shiki-light:#999999;--shiki-dark:#666666;"> &lt;-</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;"> &#39;</span><span style="--shiki-light:#B56959;--shiki-dark:#C98A7D;">data/CHIP-seq.bam</span><span style="--shiki-light:#B5695999;--shiki-dark:#C98A7D99;">&#39;</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;"> </span></span>
<span class="line"><span style="--shiki-light:#A0ADA0;--shiki-dark:#758575DD;">#fileter bam</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">filter</span><span style="--shiki-light:#999999;--shiki-dark:#666666;"> &lt;-</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;"> FilterRules</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">(</span><span style="--shiki-light:#AB5959;--shiki-dark:#CB7676;">list</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">(</span><span style="--shiki-light:#59873A;--shiki-dark:#80A665;">MinWidth</span><span style="--shiki-light:#999999;--shiki-dark:#666666;"> =</span><span style="--shiki-light:#1E754F;--shiki-dark:#4D9375;"> function</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">(</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">x</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">)</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;"> width</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">(</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">x</span><span style="--shiki-light:#1E754F;--shiki-dark:#4D9375;">$</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">seq</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">)</span><span style="--shiki-light:#AB5959;--shiki-dark:#CB7676;"> &gt;</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;"> 35</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">))</span></span>
<span class="line"><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">res</span><span style="--shiki-light:#999999;--shiki-dark:#666666;"> &lt;-</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;"> scanBam</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">(</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">test_bam_file</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">,</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;"> filter</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">=</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">filter</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">)[[</span><span style="--shiki-light:#2F798A;--shiki-dark:#4C9A91;">1</span><span style="--shiki-light:#999999;--shiki-dark:#666666;">]]</span></span>
<span class="line"><span style="--shiki-light:#998418;--shiki-dark:#B8A965;">sapply</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">(</span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">res</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">, </span><span style="--shiki-light:#B07D48;--shiki-dark:#BD976A;">head</span><span style="--shiki-light:#393A34;--shiki-dark:#DBD7CAEE;">)</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p>从上面的例子（需要在R里面运行）可以看到BAM文件需要用特殊的方法来读取，可以是R里面的Rsamtools包，也可以是linux环境下安装好的samtools软件，因为它是二进制文件，不能像普通的文本文件那样来打开。</p><p>我们用R里面的head函数查看了该BAM文件的前6行，比对的flag分别是 <code>16 0 16 16 0 0</code>,说明有3条序列没有成功比对到基因组。width信息说明该序列长度都是36bp。序列的碱基以及对应的碱基质量也如上所述。</p><h2 id="vcf" tabindex="-1"><a class="header-anchor" href="#vcf"><span>VCF</span></a></h2><p>VCF全称Variant Call Format（VCF）是一个用于存储基因序列突变信息的文本格式。 可以表示单碱基突变, 插入/缺失, 拷贝数变异和结构变异等。 通常是对BAM文件格式的比对结果进行处理得到的。 BCF格式文件是VCF格式的二进制文件。</p><p><img src="https://cdn.jsdelivr.net/gh/Moonerss/CDN/paper/filtype/vcf_variation_class.jpg" alt=""></p><p>如上图所示，vcf记录的即为各类型的变异。例如：点突变，拷贝数变异，插入，缺失等结构变异。</p><p>VCF分为两部分，注释信息和变异位点记录信息。</p><p>注释信息通常以#开头，会描述该VCF版本，目前以4.2居多，然后会一行行记录变异位点信息里面会出现的所有TAG。</p><p>下面这个是NCBI的dbSNP数据库里面的人类的vcf文件的部分截取：</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>##fileformat=VCFv4.0</span></span>
<span class="line"><span>##fileDate=20160601</span></span>
<span class="line"><span>##source=dbSNP</span></span>
<span class="line"><span>##dbSNP_BUILD_ID=147</span></span>
<span class="line"><span>##reference=GRCh37.p13</span></span>
<span class="line"><span>##phasing=partial</span></span>
<span class="line"><span>##variationPropertyDocumentationUrl=ftp://ftp.ncbi.nlm.nih.gov/snp/specs/dbSNP_BitField_latest.pdf    </span></span>
<span class="line"><span>##INFO=&lt;ID=RS,Number=1,Type=Integer,Description=&quot;dbSNP ID (i.e. rs number)&quot;&gt;</span></span>
<span class="line"><span>##INFO=&lt;ID=RSPOS,Number=1,Type=Integer,Description=&quot;Chr position reported in dbSNP&quot;&gt;</span></span>
<span class="line"><span>##INFO=&lt;ID=RV,Number=0,Type=Flag,Description=&quot;RS orientation is reversed&quot;&gt;</span></span>
<span class="line"><span></span></span>
<span class="line"><span></span></span>
<span class="line"><span>#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO</span></span>
<span class="line"><span>1       10177   rs201752861     A       C       .       .       RS=201752861;RSPOS=10177;dbSNPBuildID=137;SSR=0;SAO=0;VP=0x050000020005000002000100;GENEINFO=DDX11L1:100287102;WGT=1;VC=SNV;R5;ASP</span></span>
<span class="line"><span>1       10177   rs367896724     A       AC      .       .       RS=367896724;RSPOS=10177;dbSNPBuildID=138;SSR=0;SAO=0;VP=0x050000020005170026000200;GENEINFO=DDX11L1:100287102;WGT=1;VC=DIV;R5;ASP;VLD;G5A;G5;KGPhase3;CAF=0.5747,0.4253;COMMON=1</span></span>
<span class="line"><span>1       10228   rs143255646     TA      T       .       .       RS=143255646;RSPOS=10229;dbSNPBuildID=134;SSR=0;SAO=0;VP=0x050000020005000002000200;GENEINFO=DDX11L1:100287102;WGT=1;VC=DIV;R5;ASP</span></span>
<span class="line"><span>1       10228   rs200462216     TAACCCCTAACCCTAACCCTAAACCCTA    T       .       .       RS=200462216;RSPOS=10229;dbSNPBuildID=137;SSR=0;SAO=0;VP=0x050000020005000002000200;GENEINFO=DDX11L1:100287102;WGT=1;VC=DIV;R5;ASP</span></span>
<span class="line"><span>1       10230   rs775928745     AC      A       .       .       RS=775928745;RSPOS=10231;dbSNPBuildID=144;SSR=0;SAO=0;VP=0x050000020005000002000200;GENEINFO=DDX11L1:100287102;WGT=1;VC=DIV;R5;ASP</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p>限于文章篇幅限制，我只是截取了该VCF文件的部分注释信息，很明显可以看到注释信息刚刚开始的几行其实是没有规则的，只需要以##开头即可，描述一些必备信息，包括参考基因组版本，得到该VCF文件的命令是什么等等。</p><p>后面的都是以INFO=&lt;ID=······&gt;的形式来介绍一个个TAG，这些TAG都是会在VCF的正文，变异位点记录里面用到的。而且每个tag都很容易理解，就是对应的英文描述而已。</p><p>接下来我们看看比较复杂的正文部分，就是变异位点记录信息。</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  sample1</span></span>
<span class="line"><span>1       858691  .       TG      T       222     .       INDEL;IDV=37;IMF=0.486842;DP=76;VDB=0.110516;SGB=-0.693139;MQSB=1;MQ0F=0;ICB=1;HOB=0.5;AC=1;AN=2;DP4=12,24,14,22;MQ=60  GT:PL   0/1:255,0,255</span></span>
<span class="line"><span>1       858801  .       A       G       222     .       DP=59;VDB=0.728126;SGB=-0.692717;RPB=0.748623;MQB=1;MQSB=1;BQB=0.963908;MQ0F=0;ICB=1;HOB=0.5;AC=1;AN=2;DP4=14,17,11,12;MQ=60    GT:PL   0/1:255,0,255</span></span>
<span class="line"><span>1       859404  .       C       G       222     .       DP=81;VDB=0.0896228;SGB=-0.693132;RPB=0.849598;MQB=1;MQSB=1;BQB=0.486963;MQ0F=0;ICB=1;HOB=0.5;AC=1;AN=2;DP4=17,15,18,16;MQ=60   GT:PL   0/1:255,0,255</span></span>
<span class="line"><span>1       859690  .       C       G       222     .       DP=75;VDB=0.0662538;SGB=-0.69312;RPB=0.959181;MQB=1;MQSB=1;BQB=0.962588;MQ0F=0;ICB=1;HOB=0.5;AC=1;AN=2;DP4=20,15,18,14;MQ=60    GT:PL   0/1:255,0,255</span></span>
<span class="line"><span>1       859701  .       C       G       222     .       DP=74;VDB=0.274853;SGB=-0.693127;RPB=0.97201;MQB=1;MQSB=1;BQB=0.717302;MQ0F=0;ICB=1;HOB=0.5;AC=1;AN=2;DP4=19,15,19,14;MQ=60     GT:PL   0/1:255,0,255</span></span>
<span class="line"><span>1       859913  .       A       G       222     .       DP=67;VDB=0.756546;SGB=-0.693139;RPB=0.950685;MQB=1;MQSB=1;BQB=0.662934;MQ0F=0;ICB=1;HOB=0.5;AC=1;AN=2;DP4=18,10,19,17;MQ=60    GT:PL   0/1:255,0,255</span></span>
<span class="line"><span>1       860416  .       G       A       222     .       DP=79;VDB=0.673886;SGB=-0.693144;RPB=0.11919;MQB=1;MQSB=1;BQB=0.992984;MQ0F=0;ICB=1;HOB=0.5;AC=1;AN=2;DP4=18,15,24,15;MQ=60     GT:PL   0/1:255,0,255</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p>每一行代表一个Variant的信息。</p><ul><li><code>CHROM</code> 和 <code>POS</code>：代表参考序列名和variant的位置；如果是INDEL的话，位置是INDEL的第一个碱基位置。</li><li><code>ID</code>：variant的ID。比如在dbSNP中有该SNP的id，则会在此行给出(这个需要自己下载dbSNP数据库文件进行注释才有的)。 若没有或者注释不上，则用’.&#39;表示其为一个novel variant。</li><li><code>REF</code> 和 <code>ALT</code>：参考序列的碱基 和 Variant的碱基。</li><li><code>QUAL</code>：Phred格式(Phred_scaled)的质量值，表示在该位点存在variant的可能性；该值越高，则variant的可能性越大； 计算方法：Phred值 = -10 * log (1-p) p为variant存在的概率; 通过计算公式可以看出值为10的表示错误概率为0.1，该位点为variant的概率为90%。</li><li><code>FILTER</code>：使用上一个QUAL值来进行过滤的话，是不够的。GATK能使用其它的方法来进行过滤，过滤结果中通过则该值为 <code>PASS</code>;若variant不可靠，则该项不为 <code>PASS</code>或 <code>.</code>。</li><li><code>INFO</code>： 这一行是variant的详细信息，内容很多，以下再具体详述。</li><li><code>FORMAT</code> 和 <code>sample1</code> ：这两行合起来提供了 sample1 这个sample的基因型的信息。’sample1′代表这该名称的样品，是由SAM/BAM文件中的@RG下的 SM 标签决定的。 (当然并不是所有的VCF都是由一个BAM文件产生，比如数据库dbSNP提供的vcf文件，就没有样本信息啦)</li></ul><p><strong>1、第8列的INFO</strong></p><p>该列信息最多了，都是以 “TAG=Value”, 并使用”;”分隔的形式 。 其中 很多的注释信息在VCF文件的头部注释中给出。以下是这些TAG的解释</p><p>AC，AF 和 AN：AC(Allele Count) 表示该Allele的数目；AF(Allele Frequency) 表示Allele的频率； AN(Allele Number) 表示Allele的总数目。对于1个diploid sample而言：则基因型 0/1 表示sample为杂合子，Allele数为1(双倍体的sample在该位点只有1个等位基因发生了突变)，Allele的频率为0.5(双倍体的 sample在该位点只有50%的等位基因发生了突变)，总的Allele为2； 基因型 1/1 则表示sample为纯合的，Allele数为2，Allele的频率为1，总的Allele为2。</p><p>DP：reads覆盖度。是一些reads被过滤掉后的覆盖度。</p><p>Dels：Fraction of Reads Containing Spanning Deletions。进行SNP和INDEL calling的结果中，有该TAG并且值为0表示该位点为SNP，没有则为INDEL。</p><p>FS：使用Fisher’s精确检验来检测strand bias而得到的Fhred格式的p值。该值越小越好。一般进行filter的时候，可以设置 FS &lt; 10～20。</p><p>HaplotypeScore：Consistency of the site with at most two segregating haplotypes</p><p>InbreedingCoeff：Inbreeding coefficient as estimated from the genotype likelihoods per-sample when compared against the Hard-Weinberg expectation</p><p>MLEAC：Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed</p><p>MLEAF：Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT alle in the same order as listed</p><p>MQ：RMS Mapping Quality</p><p>MQ0：Total Mapping Quality Zero Reads</p><p>MQRankSum：Z-score From Wilcoxon rank sum test of Alt vs. Ref read mapping qualities</p><p>QD：Variant Confidence/Quality by Depth</p><p>RPA：Number of times tandem repeat unit is repeated, for each allele (including reference)</p><p>RU：Tandem repeat unit (bases)</p><p>ReadPosRankSum：Z-score from Wilcoxon rank sum test of Alt vs. Ref read position bias</p><p>STR：Variant is a short tandem repeat</p><p><strong>2、9和10列代表基因型</strong></p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>GT:AD:DP:GQ:PL 0/1:173,141:282:99:255,0,255 </span></span>
<span class="line"><span>GT:AD:DP:GQ:PL 0/1:1,3:4:25.92:103,0,26</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div></div></div><p>看上面最后两列数据，这两列数据是对应的，前者为格式，后者为格式对应的数据。这些TAG也是可以在VCF的头文件找到的</p><p>GT：样品的基因型（genotype）。两个数字中间用’/&#39;分开，这两个数字表示双倍体的sample的基因型。0 表示样品中有ref的allele； 1 表示样品中variant的allele； 2表示有第二个variant的allele。因此： 0/0 表示sample中该位点为纯合的，和ref一致； 0/1 表示sample中该位点为杂合的，有ref和variant两个基因型； 1/1 表示sample中该位点为纯合的，和variant一致。</p><p>AD 和 DP：AD(Allele Depth)为sample中每一种allele的reads覆盖度，在diploid中则是用逗号分割的两个值，前者对应ref基因型，后者对应variant基因型。</p><p>DP（Depth）为sample中该位点的测序深度。</p><p>GQ：基因型的质量值(Genotype Quality)。Phred格式(Phred_scaled)的质量值，表示在该位点该基因型存在的可能性；该值越高，则Genotype的可能性越大；计算方法：Phred值 = -10 * log (1-p) p为基因型存在的概率。</p><p>PL：指定的三种基因型的质量值(provieds the likelihoods of the given genotypes)。这三种指定的基因型为(0/0,0/1,1/1)，这三种基因型的概率总和为1。和之前不一致，该值越大，表明为该种基因型的可能性越小。 Phred值 = -10 * log (p) p为基因型存在的概率。</p><p><img src="https://cdn.jsdelivr.net/gh/Moonerss/CDN/paper/filtype/vcf_file_format.jpg" alt=""></p><h2 id="gtf和gff" tabindex="-1"><a class="header-anchor" href="#gtf和gff"><span>GTF和GFF</span></a></h2><p><strong>简介</strong></p><p>GFF全称为general feature format，这种格式主要是用来注释基因组。 GTF全称为gene transfer format，主要是用来对基因进行注释。 GTF和GFF格式是Sanger研究所定义，是一种简单的、方便的对于DNA、RNA以及蛋白质序列的特征进行描述的一种数据格式. 比如序列的哪里到哪里是基因，是转录本，是外显子，内含子或者CDS等等，已经成为序列注释的通用格式，许多软件都支持输入或者输出gff格式。</p><p><strong>一、定义和示例</strong></p><p>gff由tab键隔开的9列组成，以下是各列的说明：</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>Column 1: “seqid”</span></span>
<span class="line"><span>序列的编号，编号的有效字符[a-zA-Z0-9.:^*$@!+_?-|]</span></span>
<span class="line"><span>Column 2: “source”</span></span>
<span class="line"><span>注释信息的来源，比如”Genescan”、”Genbank” 等，可以为空，为空用”.”点号代替</span></span>
<span class="line"><span>Column 3: “type”</span></span>
<span class="line"><span>注释信息的类型，比如Gene、cDNA、mRNA等，或者是SO对应的编号</span></span>
<span class="line"><span>Columns 4 &amp; 5: “start” and “end”</span></span>
<span class="line"><span>开始与结束的位置，注意计数是从1开始的。结束位置不能大于序列的长度</span></span>
<span class="line"><span>Column 6: “score”</span></span>
<span class="line"><span>得分，数字，是注释信息可能性的说明，可以是序列相似性比对时的E-values值或者基因预测是的P-values值。”.”表示为空。</span></span>
<span class="line"><span>Column 7: “strand”</span></span>
<span class="line"><span>序列的方向， +表示正义链, -反义链 , ? 表示未知.</span></span>
<span class="line"><span>Column 8: “phase”</span></span>
<span class="line"><span>仅对注释类型为 “CDS”有效，表示起始编码的位置，有效值为0、1、2。</span></span>
<span class="line"><span>Column 9: “attributes”</span></span>
<span class="line"><span>以多个键值对组成的注释信息描述，键与值之间用”=“，不同的键值用”;“隔开，一个键可以有多个值，不同值用”,“分割。注意如果描述中包括tab键以及”,=;”，要用URL转义规则进行转义，如tab键用 %09代替。键是区分大小写的，以大写字母开头的键是预先定义好的，在后面可能被其他注释信息所调用。</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p>预先定义的键包括：</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>ID 注释信息的编号，在一个GFF文件中必须唯一；</span></span>
<span class="line"><span>Name 注释信息的名称，可以重复；</span></span>
<span class="line"><span>Alias 别名</span></span>
<span class="line"><span>Parent Indicates 该注释所属的注释，值为注释信息的编号，比如外显子所属的转录组编号，转录组所属的基因的编号。值可以为多个。</span></span>
<span class="line"><span>Target Indicates： the target of a nucleotide-to-nucleotide or protein-to-nucleotide alignment.（核苷酸对核苷酸或蛋白质至核苷酸比对的靶点。）</span></span>
<span class="line"><span>Gap：The alignment of the feature to the target if the two are not collinear (e.g. contain gaps).（如果两者不共线（例如包含间隙），则该特征与目标的对准）</span></span>
<span class="line"><span>Derives_from：Used to disambiguate the relationship between one feature and another when the relationship is a temporal one rather than a purely structural “part of” one. This is needed for polycistronic genes.（用于消除一个特征与另一个特征之间的关系，当关系是一个时间的关系，而不是纯粹的结构“一部分”时。 这是多顺反子基因所必需的。）</span></span>
<span class="line"><span>Note： 备注</span></span>
<span class="line"><span>Dbxref ：数据库索引</span></span>
<span class="line"><span>Ontology_term： A cross reference to an ontology term.</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p>下面是一个简单的实例：</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>##gff-version 3</span></span>
<span class="line"><span>##sequence-region ctg123 1 1497228</span></span>
<span class="line"><span>ctg123 . gene 1000 9000 . + . ID=gene00001;Name=EDEN</span></span>
<span class="line"><span>ctg123 . TF_binding_site 1000 1012 . + . Parent=gene00001</span></span>
<span class="line"><span>ctg123 . mRNA 1050 9000 . + . ID=mRNA00001;Parent=gene00001</span></span>
<span class="line"><span>ctg123 . mRNA 1050 9000 . + . ID=mRNA00002;Parent=gene00001 ctg123 . mRNA 1300 9000 . + . ID=mRNA00003;Parent=gene00001 ctg123 . exon 1300 1500 . + . Parent=mRNA00003</span></span>
<span class="line"><span>ctg123 . exon 1050 1500 . + . Parent=mRNA00001,mRNA00002</span></span>
<span class="line"><span>ctg123 . exon 3000 3902 . + . Parent=mRNA00001,mRNA00003</span></span>
<span class="line"><span>ctg123 . exon 5000 5500 . + . Parent=mRNA00001,mRNA00002,mRNA00003</span></span>
<span class="line"><span>ctg123 . exon 7000 9000 . + . Parent=mRNA00001,mRNA00002,mRNA00003</span></span>
<span class="line"><span>ctg123 . CDS 1201 1500 . + 0 ID=cds00001;Parent=mRNA00001</span></span>
<span class="line"><span>ctg123 . CDS 3000 3902 . + 0 ID=cds00001;Parent=mRNA00001</span></span>
<span class="line"><span>ctg123 . CDS 5000 5500 . + 0 ID=cds00001;Parent=mRNA00001</span></span>
<span class="line"><span>ctg123 . CDS 7000 7600 . + 0 ID=cds00001;Parent=mRNA00001</span></span>
<span class="line"><span>ctg123 . CDS 1201 1500 . + 0 ID=cds00002;Parent=mRNA00002</span></span>
<span class="line"><span>ctg123 . CDS 5000 5500 . + 0 ID=cds00002;Parent=mRNA00002</span></span>
<span class="line"><span>ctg123 . CDS 7000 7600 . + 0 ID=cds00002;Parent=mRNA00002</span></span>
<span class="line"><span>ctg123 . CDS 3301 3902 . + 0 ID=cds00003;Parent=mRNA00003</span></span>
<span class="line"><span>ctg123 . CDS 5000 5500 . + 2 ID=cds00003;Parent=mRNA00003</span></span>
<span class="line"><span>ctg123 . CDS 7000 7600 . + 2 ID=cds00003;Parent=mRNA00003</span></span>
<span class="line"><span>ctg123 . CDS 3391 3902 . + 0 ID=cds00004;Parent=mRNA00003</span></span>
<span class="line"><span>ctg123 . CDS 5000 5500 . + 2 ID=cds00004;Parent=mRNA00003</span></span>
<span class="line"><span>ctg123 . CDS 7000 7600 . + 2 ID=cds00004;Parent=mRNA00003</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p>可以看到第9列其实是可以无限扩展的，只需要以封号进行分割即可。</p><p><strong>二、GTF 与GFF的差异</strong></p><p>GTF文件以及GFF文件都由9列数据组成，这两种文件的前8列都是相同的（一些小的差别），它们的差别重点在第9列。</p><p>GTF文件的第9列同GFF文件不同，虽然同样是标签与值配对的情况，但标签与值之间以空格分开,而不是GFF里面的=号 且每个特征之后都要有分号，（包括最后一个特征）. 下面看一个GTF的实例：</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>17	havana	five_prime_utr	7687377	7687427	.	-	.	gene_id &quot;ENSG00000141510&quot;; gene_version &quot;16&quot;; transcript_id &quot;ENST00000503591&quot;; transcript_version &quot;1&quot;; gene_name &quot;TP53&quot;; gene_source &quot;ensembl_havana&quot;; gene_biotype &quot;protein_coding&quot;; havana_gene &quot;OTTHUMG00000162125&quot;; havana_gene_version &quot;10&quot;; transcript_name &quot;TP53-003&quot;; transcript_source &quot;havana&quot;; transcript_biotype &quot;protein_coding&quot;; havana_transcript &quot;OTTHUMT00000367399&quot;; havana_transcript_version &quot;2&quot;; tag &quot;cds_end_NF&quot;; tag &quot;mRNA_end_NF&quot;; transcript_support_level &quot;5&quot;;</span></span>
<span class="line"><span>17	havana	five_prime_utr	7677325	7677427	.	-	.	gene_id &quot;ENSG00000141510&quot;; gene_version &quot;16&quot;; transcript_id &quot;ENST00000503591&quot;; transcript_version &quot;1&quot;; gene_name &quot;TP53&quot;; gene_source &quot;ensembl_havana&quot;; gene_biotype &quot;protein_coding&quot;; havana_gene &quot;OTTHUMG00000162125&quot;; havana_gene_version &quot;10&quot;; transcript_name &quot;TP53-003&quot;; transcript_source &quot;havana&quot;; transcript_biotype &quot;protein_coding&quot;; havana_transcript &quot;OTTHUMT00000367399&quot;; havana_transcript_version &quot;2&quot;; tag &quot;cds_end_NF&quot;; tag &quot;mRNA_end_NF&quot;; transcript_support_level &quot;5&quot;;</span></span>
<span class="line"><span>17	havana	five_prime_utr	7676595	7676622	.	-	.	gene_id &quot;ENSG00000141510&quot;; gene_version &quot;16&quot;; transcript_id &quot;ENST00000503591&quot;; transcript_version &quot;1&quot;; gene_name &quot;TP53&quot;; gene_source &quot;ensembl_havana&quot;; gene_biotype &quot;protein_coding&quot;; havana_gene &quot;OTTHUMG00000162125&quot;; havana_gene_version &quot;10&quot;; transcript_name &quot;TP53-003&quot;; transcript_source &quot;havana&quot;; transcript_biotype &quot;protein_coding&quot;; havana_transcript &quot;OTTHUMT00000367399&quot;; havana_transcript_version &quot;2&quot;; tag &quot;cds_end_NF&quot;; tag &quot;mRNA_end_NF&quot;; transcript_support_level &quot;5&quot;;</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><h2 id="genepred" tabindex="-1"><a class="header-anchor" href="#genepred"><span>GenePred</span></a></h2><p><strong>简介</strong></p><p>GenePred格式主要用于基因浏览器中基因预测的track。 如果有可变剪切的情况，那表格的每一行就是一个 transcript 的全部信息。</p><p><strong>一、定义和示例</strong></p><p>每一行的具体解释如下</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>table genePred</span></span>
<span class="line"><span>&quot;A gene prediction.&quot;</span></span>
<span class="line"><span>    (</span></span>
<span class="line"><span>    string  name;               &quot;Name of gene&quot;</span></span>
<span class="line"><span>    string  chrom;              &quot;Chromosome name&quot;</span></span>
<span class="line"><span>    char[1] strand;             &quot;+ or - for strand&quot;</span></span>
<span class="line"><span>    uint    txStart;            &quot;Transcription start position&quot;</span></span>
<span class="line"><span>    uint    txEnd;              &quot;Transcription end position&quot;</span></span>
<span class="line"><span>    uint    cdsStart;           &quot;Coding region start&quot;</span></span>
<span class="line"><span>    uint    cdsEnd;             &quot;Coding region end&quot;</span></span>
<span class="line"><span>    uint    exonCount;          &quot;Number of exons&quot;</span></span>
<span class="line"><span>    uint[exonCount] exonStarts; &quot;Exon start positions&quot;</span></span>
<span class="line"><span>    uint[exonCount] exonEnds;   &quot;Exon end positions&quot;</span></span>
<span class="line"><span>    )</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p>如果觉得抽象，我们可以用示例来进行一下对比。小编在这里首先将模式植物拟南芥的gtf文件转化为gpd格式。<code>head -n 1</code> 看一下gpd文件第一行的样式。</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>#gpd</span></span>
<span class="line"><span>AT1G01010.1	Chr1	+	3630	5899	3759	5630	6	3630,3995,4485,4705,5173,5438,	3913,4276,4605,5095,5326,5899</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div></div></div><p>再来 <code>grep</code>一下gtf文件中有<strong>AT1G01010.1</strong>信息的那些行是什么样</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>#gtf</span></span>
<span class="line"><span>Chr1	Araport11	5UTR	3631	3759	.	+	.	transcript_id &quot;AT1G01010.1&quot;; gene_id &quot;AT1G01010&quot;</span></span>
<span class="line"><span>Chr1	Araport11	exon	3631	3913	.	+	.	transcript_id &quot;AT1G01010.1&quot;; gene_id &quot;AT1G01010&quot;</span></span>
<span class="line"><span>Chr1	Araport11	start_codon	3760	3762	.	+	.	transcript_id &quot;AT1G01010.1&quot;; gene_id &quot;AT1G01010&quot;</span></span>
<span class="line"><span>Chr1	Araport11	CDS	3760	3913	.	+	0	transcript_id &quot;AT1G01010.1&quot;; gene_id &quot;AT1G01010&quot;</span></span>
<span class="line"><span>Chr1	Araport11	exon	3996	4276	.	+	.	transcript_id &quot;AT1G01010.1&quot;; gene_id &quot;AT1G01010&quot;</span></span>
<span class="line"><span>Chr1	Araport11	CDS	3996	4276	.	+	2	transcript_id &quot;AT1G01010.1&quot;; gene_id &quot;AT1G01010&quot;</span></span>
<span class="line"><span>Chr1	Araport11	exon	4486	4605	.	+	.	transcript_id &quot;AT1G01010.1&quot;; gene_id &quot;AT1G01010&quot;</span></span>
<span class="line"><span>Chr1	Araport11	CDS	4486	4605	.	+	0	transcript_id &quot;AT1G01010.1&quot;; gene_id &quot;AT1G01010&quot;</span></span>
<span class="line"><span>Chr1	Araport11	exon	4706	5095	.	+	.	transcript_id &quot;AT1G01010.1&quot;; gene_id &quot;AT1G01010&quot;</span></span>
<span class="line"><span>Chr1	Araport11	CDS	4706	5095	.	+	0	transcript_id &quot;AT1G01010.1&quot;; gene_id &quot;AT1G01010&quot;</span></span>
<span class="line"><span>Chr1	Araport11	exon	5174	5326	.	+	.	transcript_id &quot;AT1G01010.1&quot;; gene_id &quot;AT1G01010&quot;</span></span>
<span class="line"><span>Chr1	Araport11	CDS	5174	5326	.	+	0	transcript_id &quot;AT1G01010.1&quot;; gene_id &quot;AT1G01010&quot;</span></span>
<span class="line"><span>Chr1	Araport11	CDS	5439	5630	.	+	0	transcript_id &quot;AT1G01010.1&quot;; gene_id &quot;AT1G01010&quot;</span></span>
<span class="line"><span>Chr1	Araport11	exon	5439	5899	.	+	.	transcript_id &quot;AT1G01010.1&quot;; gene_id &quot;AT1G01010&quot;</span></span>
<span class="line"><span>Chr1	Araport11	stop_codon	5628	5630	.	+	.	transcript_id &quot;AT1G01010.1&quot;; gene_id &quot;AT1G01010&quot;</span></span>
<span class="line"><span>Chr1	Araport11	3UTR	5631	5899	.	+	.	transcript_id &quot;AT1G01010.1&quot;; gene_id &quot;AT1G01010&quot;</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p>我们可以看出，在GTF中，AT1G01010.1这个transcript共有6个CDS，那么对应到相应gpd文件AT1G01010.1这一行的第8列就是6，而第9列和第10列则是这6个CDS对应的起始和终止位置。</p><p>细心的朋友可能会发现，GTF文件中CDS起始位置在GenePred table中统统少了1，这其实就是两种文件的起始位置从1开始还是从0开始计数的区别。</p><p>GTF的产生和流行有其历史的原因。但是从技术角度来讲，这个文件格式是个非常差劲的格式。</p><p>GTF格式非常冗余。以人类转录组为例，Gencode V22的GTF文件为1.2G，压缩之后只有40M。大家知道压缩软件的压缩比是和软件的冗余程度。很少有文件能够压缩到1/30的大小。可见GTF格式多么冗余。GTF格式（及其早期版本GFF等）有很好的替代格式。从信息量上来讲：GTF 等价于 GenePred （Bed12) + Gene_Anno_table。</p><p>GenePred是Jimmy Kent创建UCSC genome browser的时候建立的文件格式。UCSC的文件格式定义是非常smart的，包括之后我可能会讲到的2bit，bigwig格式。</p><p><strong>二、GTF vs GenePred</strong></p><ul><li>从文件大小上来看，压缩前：GTF（1.2G) &gt;&gt; Genepred(23M) + Gene_Anno_table (2.8M)。压缩后：GTF(40M) &gt;&gt; GenePred(7.8M) +Gene_Anno_table (580K)</li><li>从可读性来讲，GTF是以gene interval 为单位（行），每行可以是gene，transcript，exon，intron，utr等各种信息，看起来什么都在里面，很全面，其实可读性非常差，而且容易产生各种错误。GenePred格式是以transcript为单位，每行就是一个transcript，简洁直观。</li><li>从程序处理的角度来讲：以GTF文件作为输入的程序，如果换成以GenePred格式为输入，编程的难度会降低一个数量级，运算时间会快很多，代码的可读性强很多。</li></ul><p>GTF 转换成GenePred：</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>gtfToGenePred -genePredExt -ignoreGroupsWithoutExons -geneNameAsName2 test.gtf test.gpd</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div></div></div><p>Gene_Anno_table: 其实就是把GTF的所有transcript行的第9列转换变成一个表格。</p><h2 id="bed" tabindex="-1"><a class="header-anchor" href="#bed"><span>BED</span></a></h2><p><strong>简介</strong></p><p>BED 文件格式提供了一种灵活的方式来定义的数据行，用于描述注释的信息。 跟GTF/GFF格式一样，也可以用来描述基因组特征。但没有GTF/GFF格式那么正规，通常用来描述 任何人为定义的区间。 但没有GTF/GFF格式那么正规，通常用来描述任何人为定义的区间。 所以BED格式最重要的就是染色体加上起始终止坐标这3列。</p><p><strong>一、定义和示例</strong></p><p>BED行有3个必须的列和9个额外可选的列。 每行的数据格式要求一致。</p><p><strong>1、必须包含的3列是：</strong></p><ol><li>chrom, 染色体或scafflold 的名字(eg chr3， chrY, chr2_random, scaffold0671 )</li><li>chromStart 染色体或scaffold的起始位置，染色体第一个碱基的位置是0</li><li>chromEnd 染色体或scaffold的结束位置，染色体的末端位置没有包含到显示信息里面。例如，首先得100个碱基的染色体定义为chromStart =0 . chromEnd=100, 碱基的数目是0-99</li></ol><p><strong>2、9 个额外的可选列:</strong></p><ol><li>name 指定BED行的名字，这个名字标签会展示在基因组浏览器中的bed行的左侧。</li><li>score 0到1000的分值，如果在注释数据的设定中将原始基线设置为１，那么这个分值会决定现示灰度水平（数字越大，灰度越高）</li><li>strand 定义链的方向，&#39;&#39;+” 或者”-”</li><li>thickStart 起始位置（The starting position at which the feature is drawn thickly）(例如，基因起始编码位置）</li><li>thickEnd 终止位置（The ending position at which the feature is drawn thickly）（例如：基因终止编码位置）</li><li>itemRGB 是一个RGB值的形式, R, G, B (eg. 255, 0, 0), 如果itemRgb设置为&#39;On”, 这个RBG值将决定数据的显示的颜色。</li><li>blockCount BED行中的block数目，也就是外显子数目</li><li>blockSize 用逗号分割的外显子的大小,这个item的数目对应于BlockCount的数目</li><li>blockStarts- 用逗号分割的列表, 所有外显子的起始位置，数目也与blockCount数目对应.</li></ol><p><strong>3、一个简单的示例如下：</strong></p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>track name=pairedReads description=&quot;Clone Paired Reads&quot; useScore=1</span></span>
<span class="line"><span>chr22 1000 5000 cloneA 960 + 1000 5000 0 2 567,488, 0,3512</span></span>
<span class="line"><span>chr22 2000 6000 cloneB 900 - 2000 6000 0 2 433,399, 0,3601</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p>bed格式有相应的软件来处理这类格式的文件，如bedtools。</p><ul><li><p>注意：用于在GBrowse上展示相关注释的bed格式通常第一行有一个关于track的描述信息。</p><p><strong>速记：</strong></p></li><li><p>bed不是床，缺了主要3列就得黄~</p></li><li><p>9列可选列，看了不会胡略略~</p></li></ul><p><strong>二、BED 与GFF的差异</strong></p><p>BED文件中起始坐标为0，结束坐标至少是1,</p><p>GFF中起始坐标是1而结束坐标至少是1。</p><h2 id="maf" tabindex="-1"><a class="header-anchor" href="#maf"><span>MAF</span></a></h2><p><strong>简介</strong></p><p>MAF格式通常用于记录体细胞突变。 MAF本来并不是一个常见的文本文件格式，只是因为癌症研究实在是太热门了，对它的理解也变得需求旺盛起来了。</p><p><strong>一、MAF的说明</strong></p><p>这些文件应该使用下面描述的突变注释格式（MAF）进行格式化。另外下文中有文件命名规范。</p><p>以下几种类型的体细胞突变会在MAF文件中出现：</p><ul><li>错义突变及无义突变 *剪接位点，其定义为剪接位点2 bp以内的SNP *沉默突变 *与基因的编码区、剪接位点或遗传元件目标区域重叠的引物。 *移码突变 *调控区突变</li></ul><p>大部分MAF提交提交的是原始数据。这些原始数据中在体细胞中标记的位点与已知的变异类型相重合的。为避免有可能出现的细胞系污染，MAF规定了一定的下细胞过滤标准。根据现行政策，可开放获取MAF资料应满足：</p><ul><li>包括所有已验证的体细胞突变名称 *包括与编码区域或剪接位点重叠的所有未验证的体细胞突变名称 *排除所有其他类型的突变（即非体细胞突变、不在编码区域或剪接位点的未验证体的细胞突变以及未在dbSNP、COSMIC或OMIM中注释为体细胞的dbSNP位点）</li></ul><p>我们提交给DCC MAF存档的数据包括两种：Somatic MAF（named .somatic.maf）的开放访问数据以及不经过筛选的包含原始数据的Protected MAF（named.protected.maf）。所有数据将使用MAF标准进行格式化。</p><p><strong>二、MAF文件字段</strong></p><p>MAF文件的格式是以制表符分隔的列。这些列都在表1中进行了详细注释，每个MAF文件中都必须按照相应格式进行处理，DCC将验证每列的顺序是否符合标注。每列的标题和值有时需要区分大小写。列中允许出现空值（即空白单元格）或枚举值。验证器将查找是否存在一个符合规范（例如#version 2.4）的标题，如果没有，验证将会失败。除了出现在表1中的列外，也可以选择附加其他列。可选列不经过DCC验证，可以按任何顺序进行。</p><p>MAF文件可能有两种格式 ，可能是47列，或者120列，第一行一般都是 头文件，注释着每一列的信息，的确，信息量有点略大。如下：</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>     1	Hugo_Symbol</span></span>
<span class="line"><span>     2	Entrez_Gene_Id</span></span>
<span class="line"><span>     3	Center</span></span>
<span class="line"><span>     4	NCBI_Build</span></span>
<span class="line"><span>     5	Chromosome</span></span>
<span class="line"><span>     6	Start_Position</span></span>
<span class="line"><span>     7	End_Position</span></span>
<span class="line"><span>     8	Strand</span></span>
<span class="line"><span>     9	Consequence</span></span>
<span class="line"><span>    10	Variant_Classification</span></span>
<span class="line"><span>    11	Variant_Type</span></span>
<span class="line"><span>    12	Reference_Allele</span></span>
<span class="line"><span>    13	Tumor_Seq_Allele1</span></span>
<span class="line"><span>    14	Tumor_Seq_Allele2</span></span>
<span class="line"><span>    15	dbSNP_RS</span></span>
<span class="line"><span>    16	dbSNP_Val_Status</span></span>
<span class="line"><span>    17	Tumor_Sample_Barcode</span></span>
<span class="line"><span>    18	Matched_Norm_Sample_Barcode</span></span>
<span class="line"><span>    19	Match_Norm_Seq_Allele1</span></span>
<span class="line"><span>    20	Match_Norm_Seq_Allele2</span></span>
<span class="line"><span>    21	Tumor_Validation_Allele1</span></span>
<span class="line"><span>    22	Tumor_Validation_Allele2</span></span>
<span class="line"><span>    23	Match_Norm_Validation_Allele1</span></span>
<span class="line"><span>    24	Match_Norm_Validation_Allele2</span></span>
<span class="line"><span>    25	Verification_Status</span></span>
<span class="line"><span>    26	Validation_Status</span></span>
<span class="line"><span>    27	Mutation_Status</span></span>
<span class="line"><span>    28	Sequencing_Phase</span></span>
<span class="line"><span>    29	Sequence_Source</span></span>
<span class="line"><span>    30	Validation_Method</span></span>
<span class="line"><span>    31	Score</span></span>
<span class="line"><span>    32	BAM_File</span></span>
<span class="line"><span>    33	Sequencer</span></span>
<span class="line"><span>    34	t_ref_count</span></span>
<span class="line"><span>    35	t_alt_count</span></span>
<span class="line"><span>    36	n_ref_count</span></span>
<span class="line"><span>    37	n_alt_count</span></span>
<span class="line"><span>    38	HGVSc</span></span>
<span class="line"><span>    39	HGVSp</span></span>
<span class="line"><span>    40	HGVSp_Short</span></span>
<span class="line"><span>    41	Transcript_ID</span></span>
<span class="line"><span>    42	RefSeq</span></span>
<span class="line"><span>    43	Protein_position</span></span>
<span class="line"><span>    44	Codons</span></span>
<span class="line"><span>    45	Hotspot</span></span>
<span class="line"><span>    46	cDNA_change</span></span>
<span class="line"><span>    47	Amino_Acid_Change</span></span>
<span class="line"><span>     1	Hugo_Symbol</span></span>
<span class="line"><span>     2	Entrez_Gene_Id</span></span>
<span class="line"><span>     3	Center</span></span>
<span class="line"><span>     4	NCBI_Build</span></span>
<span class="line"><span>     5	Chromosome</span></span>
<span class="line"><span>     6	Start_Position</span></span>
<span class="line"><span>     7	End_Position</span></span>
<span class="line"><span>     8	Strand</span></span>
<span class="line"><span>     9	Variant_Classification</span></span>
<span class="line"><span>    10	Variant_Type</span></span>
<span class="line"><span>    11	Reference_Allele</span></span>
<span class="line"><span>    12	Tumor_Seq_Allele1</span></span>
<span class="line"><span>    13	Tumor_Seq_Allele2</span></span>
<span class="line"><span>    14	dbSNP_RS</span></span>
<span class="line"><span>    15	dbSNP_Val_Status</span></span>
<span class="line"><span>    16	Tumor_Sample_Barcode</span></span>
<span class="line"><span>    17	Matched_Norm_Sample_Barcode</span></span>
<span class="line"><span>    18	Match_Norm_Seq_Allele1</span></span>
<span class="line"><span>    19	Match_Norm_Seq_Allele2</span></span>
<span class="line"><span>    20	Tumor_Validation_Allele1</span></span>
<span class="line"><span>    21	Tumor_Validation_Allele2</span></span>
<span class="line"><span>    22	Match_Norm_Validation_Allele1</span></span>
<span class="line"><span>    23	Match_Norm_Validation_Allele2</span></span>
<span class="line"><span>    24	Verification_Status</span></span>
<span class="line"><span>    25	Validation_Status</span></span>
<span class="line"><span>    26	Mutation_Status</span></span>
<span class="line"><span>    27	Sequencing_Phase</span></span>
<span class="line"><span>    28	Sequence_Source</span></span>
<span class="line"><span>    29	Validation_Method</span></span>
<span class="line"><span>    30	Score</span></span>
<span class="line"><span>    31	BAM_File</span></span>
<span class="line"><span>    32	Sequencer</span></span>
<span class="line"><span>    33	Tumor_Sample_UUID</span></span>
<span class="line"><span>    34	Matched_Norm_Sample_UUID</span></span>
<span class="line"><span>    35	HGVSc</span></span>
<span class="line"><span>    36	HGVSp</span></span>
<span class="line"><span>    37	HGVSp_Short</span></span>
<span class="line"><span>    38	Transcript_ID</span></span>
<span class="line"><span>    39	Exon_Number</span></span>
<span class="line"><span>    40	t_depth</span></span>
<span class="line"><span>    41	t_ref_count</span></span>
<span class="line"><span>    42	t_alt_count</span></span>
<span class="line"><span>    43	n_depth</span></span>
<span class="line"><span>    44	n_ref_count</span></span>
<span class="line"><span>    45	n_alt_count</span></span>
<span class="line"><span>    46	all_effects</span></span>
<span class="line"><span>    47	Allele</span></span>
<span class="line"><span>    48	Gene</span></span>
<span class="line"><span>    49	Feature</span></span>
<span class="line"><span>    50	Feature_type</span></span>
<span class="line"><span>    51	One_Consequence</span></span>
<span class="line"><span>    52	Consequence</span></span>
<span class="line"><span>    53	cDNA_position</span></span>
<span class="line"><span>    54	CDS_position</span></span>
<span class="line"><span>    55	Protein_position</span></span>
<span class="line"><span>    56	Amino_acids</span></span>
<span class="line"><span>    57	Codons</span></span>
<span class="line"><span>    58	Existing_variation</span></span>
<span class="line"><span>    59	ALLELE_NUM</span></span>
<span class="line"><span>    60	DISTANCE</span></span>
<span class="line"><span>    61	TRANSCRIPT_STRAND</span></span>
<span class="line"><span>    62	SYMBOL</span></span>
<span class="line"><span>    63	SYMBOL_SOURCE</span></span>
<span class="line"><span>    64	HGNC_ID</span></span>
<span class="line"><span>    65	BIOTYPE</span></span>
<span class="line"><span>    66	CANONICAL</span></span>
<span class="line"><span>    67	CCDS</span></span>
<span class="line"><span>    68	ENSP</span></span>
<span class="line"><span>    69	SWISSPROT</span></span>
<span class="line"><span>    70	TREMBL</span></span>
<span class="line"><span>    71	UNIPARC</span></span>
<span class="line"><span>    72	RefSeq</span></span>
<span class="line"><span>    73	SIFT</span></span>
<span class="line"><span>    74	PolyPhen</span></span>
<span class="line"><span>    75	EXON</span></span>
<span class="line"><span>    76	INTRON</span></span>
<span class="line"><span>    77	DOMAINS</span></span>
<span class="line"><span>    78	GMAF</span></span>
<span class="line"><span>    79	AFR_MAF</span></span>
<span class="line"><span>    80	AMR_MAF</span></span>
<span class="line"><span>    81	ASN_MAF</span></span>
<span class="line"><span>    82	EAS_MAF</span></span>
<span class="line"><span>    83	EUR_MAF</span></span>
<span class="line"><span>    84	SAS_MAF</span></span>
<span class="line"><span>    85	AA_MAF</span></span>
<span class="line"><span>    86	EA_MAF</span></span>
<span class="line"><span>    87	CLIN_SIG</span></span>
<span class="line"><span>    88	SOMATIC</span></span>
<span class="line"><span>    89	PUBMED</span></span>
<span class="line"><span>    90	MOTIF_NAME</span></span>
<span class="line"><span>    91	MOTIF_POS</span></span>
<span class="line"><span>    92	HIGH_INF_POS</span></span>
<span class="line"><span>    93	MOTIF_SCORE_CHANGE</span></span>
<span class="line"><span>    94	IMPACT</span></span>
<span class="line"><span>    95	PICK</span></span>
<span class="line"><span>    96	VARIANT_CLASS</span></span>
<span class="line"><span>    97	TSL</span></span>
<span class="line"><span>    98	HGVS_OFFSET</span></span>
<span class="line"><span>    99	PHENO</span></span>
<span class="line"><span>   100	MINIMISED</span></span>
<span class="line"><span>   101	ExAC_AF</span></span>
<span class="line"><span>   102	ExAC_AF_Adj</span></span>
<span class="line"><span>   103	ExAC_AF_AFR</span></span>
<span class="line"><span>   104	ExAC_AF_AMR</span></span>
<span class="line"><span>   105	ExAC_AF_EAS</span></span>
<span class="line"><span>   106	ExAC_AF_FIN</span></span>
<span class="line"><span>   107	ExAC_AF_NFE</span></span>
<span class="line"><span>   108	ExAC_AF_OTH</span></span>
<span class="line"><span>   109	ExAC_AF_SAS</span></span>
<span class="line"><span>   110	GENE_PHENO</span></span>
<span class="line"><span>   111	FILTER</span></span>
<span class="line"><span>   112	CONTEXT</span></span>
<span class="line"><span>   113	src_vcf_id</span></span>
<span class="line"><span>   114	tumor_bam_uuid</span></span>
<span class="line"><span>   115	normal_bam_uuid</span></span>
<span class="line"><span>   116	case_id</span></span>
<span class="line"><span>   117	GDC_FILTER</span></span>
<span class="line"><span>   118	COSMIC</span></span>
<span class="line"><span>   119	MC3_Overlap</span></span>
<span class="line"><span>   120	GDC_Validation_Status</span></span>
<span class="line"><span>重要标准</span></span>
<span class="line"><span>列表中每列的顺序最好与索引列相同。</span></span>
<span class="line"><span>标有“Case Sensitive“的列所有标题都需要区分大小写。</span></span>
<span class="line"><span>标有”Null“的列表示允许具有空值。</span></span>
<span class="line"><span>标有“Enumerated column”的列表示具有指定的值，比如“Enumerated value” 是&quot;No&quot;表示该列没有指定的值；其他值表示允许列出的具体值; “Set”表示该列的值来自指定的一组已知值（例如HUGO基因符号）</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p><strong>三、MAF文件检查</strong></p><p>DCC 档案验证器将检查MAF文件的完整性。如果MAF文件中的任何一项出现错误，验证将失败：</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>1.	列标题文本（包括大小写）和顺序必须与表1完全一致</span></span>
<span class="line"><span>2.	表1中列出的列标题下的值不是空值时必须具有相应的值</span></span>
<span class="line"><span>3.	表1中指定为“Case Sensitive”的值必须区分大小写。</span></span>
<span class="line"><span>4.	如果列标题在规范中列出为具有枚举值（即“Enumerated”列的“Yes”），则这些列中的值必须来自“Enumerated”下列出的枚举值。</span></span>
<span class="line"><span>5.	如果规范中列标题具有设置值（即“Enumerated”列的“Set”），则那些列下的值必须来自该域的枚举值（例如，HUGO基因符号）。</span></span>
<span class="line"><span>6.	所有Allele-based列必须包含 –(deletion)或由以下大写字母组成的字符串：A，T，G，C。</span></span>
<span class="line"><span>7.	如果Validation_Status == &quot;Untested&quot; 那么Tumor_Validation_Allele1, Tumor_Validation_Allele2, Match_Norm_Validation_Allele1, Match_Norm_Validation_Allele2 可以是空值(由 Validation_Status决定).</span></span>
<span class="line"><span>a)	如果Validation_Status == &quot;Inconclusive&quot; 那么Tumor_Validation_Allele1, Tumor_Validation_Allele2, Match_Norm_Validation_Allele1, Match_Norm_Validation_Allele2 可以是空值(由 Validation_Status决定)</span></span>
<span class="line"><span>8.	如果Validation_Status == Valid, 那么Validated_Tumor_Allele1 and Validated_Tumor_Allele2一定要是 A, C, G, T, 或“-“中的一种</span></span>
<span class="line"><span></span></span>
<span class="line"><span>a)	如果Validation_Status  == &quot;Valid&quot; 那么Tumor_Validation_Allele1, Tumor_Validation_Allele2, Match_Norm_Validation_Allele1, Match_Norm_Validation_Allele2 不可以是空值</span></span>
<span class="line"><span></span></span>
<span class="line"><span>b)	 如果Validation_Status == &quot;Invalid&quot; 那么Tumor_Validation_Allele1, Tumor_Validation_Allele2, Match_Norm_Validation_Allele1, Match_Norm_Validation_Allele2 不可以是空值Tumor_Validation_Allelle1 == Match_Norm_Validation_Allele1 </span></span>
<span class="line"><span>Tumor_Validation_Allelle2 == Match_Norm_Validation_Allele2  (出现错误时，增加以替代8a)</span></span>
<span class="line"><span></span></span>
<span class="line"><span>9.	检查Mutation_Status的等位基因值:</span></span>
<span class="line"><span>检查Validation_status的等位基因值:</span></span>
<span class="line"><span></span></span>
<span class="line"><span>a)	如果Mutation_Status == &quot;Germline&quot; and Validation_Status == &quot;Valid&quot;, 那么Tumor_Validation_Allele1 == Match_Norm_Validation_Allele1 Tumor_Validation_Allele2 == Match_Norm_Validation_Allele2.</span></span>
<span class="line"><span></span></span>
<span class="line"><span>b)	如果Mutation_Status == &quot;Somatic&quot; ，Validation_Status == &quot;Valid&quot;, 那么Match_Norm_Validation_Allele1 == Match_Norm_Validation_Allele2 == Reference_Allele 且(Tumor_Validation_Allele1 or Tumor_Validation_Allele2) != Reference_Allele</span></span>
<span class="line"><span></span></span>
<span class="line"><span>c)	如果Mutation_Status == &quot;LOH&quot; and Validation_Status==&quot;Valid&quot;, 那么Tumor_Validation_Allele1 == Tumor_Validation_Allele2 Match_Norm_Validation_Allele1 != Match_Norm_Validation_Allele2 and Tumor_Validation_Allele1 == (Match_Norm_Validation_Allele1 or Match_Norm_Validation_Allele2).</span></span>
<span class="line"><span>10.	检查 Start_position &lt;= End_position</span></span>
<span class="line"><span>11.	根据Variant_Type检查Start_position和End_position: </span></span>
<span class="line"><span>a)	如果Variant_Type == &quot;INS&quot;, 那么(End_position - Start_position + 1 == length (Reference_Allele) 或End_position - Start_position == 1) 且length(Reference_Allele) &lt;= length(Tumor_Seq_Allele1 and Tumor_Seq_Allele2)</span></span>
<span class="line"><span>b)	如果Variant_Type == &quot;DEL&quot;, 那么End_position - Start_position + 1 == length (Reference_Allele), </span></span>
<span class="line"><span>且length(Reference_Allele) &gt;= length(Tumor_Seq_Allele1 and Tumor_Seq_Allele2)</span></span>
<span class="line"><span>c)	如果Variant_Type == &quot;SNP&quot;, 那么length(Reference_Allele and Tumor_Seq_Allele1 and Tumor_Seq_Allele2) ==  1 且(Reference_Allele and Tumor_Seq_Allele1 and Tumor_Seq_Allele2) != &quot;-&quot;</span></span>
<span class="line"><span>d)	如果Variant_Type == &quot;DNP&quot;, 那么length(Reference_Allele 和Tumor_Seq_Allele1 and Tumor_Seq_Allele2) ==  2 且(Reference_Allele and Tumor_Seq_Allele1 andTumor_Seq_Allele2) !contain &quot;-&quot;</span></span>
<span class="line"><span>e)	如果Variant_Type == &quot;TNP&quot;, 那么length(Reference_Allele and Tumor_Seq_Allele1 and Tumor_Seq_Allele2) ==  3 且(Reference_Allele and Tumor_Seq_Allele1 and Tumor_Seq_Allele2) !contain &quot;-&quot;</span></span>
<span class="line"><span>f)	如果Variant_Type == &quot;ONP&quot;, 那么length(Reference_Allele) == length(Tumor_Seq_Allele1) == length(Tumor_Seq_Allele2) &gt; 3 且(Reference_Allele and Tumor_Seq_Allele1 and Tumor_Seq_Allele2) !contain &quot;-&quot;</span></span>
<span class="line"><span>12.	基于UUID的文件的验证: </span></span>
<span class="line"><span>a)	列＃33必须包含肿瘤样本的BCR等分试样的UUID的Tumor_Sample_UUID</span></span>
<span class="line"><span>b)	列＃34必须是Matched_Norm_Sample_UUID，其中包含用于匹配正常样本的BCR等份的UUID</span></span>
<span class="line"><span>c)	由Tumor_Sample_Barcode和Matched_Norm_Sample_Barcode表示的元数据应分别对应于分配给Tumor_Sample_UUID和Matched_Norm_Sample_UUID的UUID</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p><strong>三、MAF命名协议</strong></p><p>在上传到DCC的档案中，MAF文件名应与以下方式与包含档案名称相关： 如果存档名称是：</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>&lt;domain&gt;_&lt;disease_abbrev&gt;.&lt;platform&gt;.Level_2.&lt;serial_index&gt;.&lt;revision&gt;.0.tar.gz</span></span>
<span class="line"><span>那么具有存档的公开的MAF文件命名依据为：</span></span>
<span class="line"><span>&lt;domain&gt;_&lt;disease_abbrev&gt;.&lt;platform&gt;.Level_2.&lt;serial_index&gt;[.&lt;optional_tag&gt;].somatic.maf</span></span>
<span class="line"><span>具有存档的受保护的MAF文件命名依据为：</span></span>
<span class="line"><span>&lt;domain&gt;_&lt;disease_abbrev&gt;.&lt;platform&gt;.Level_2.&lt;serial_index&gt;[.&lt;optional_tag&gt;].protected.maf</span></span>
<span class="line"><span>&lt;optional_tag&gt;可能包含字母数字字符、破折号或下划线，但不能有空格或句号。&lt;optional_tag&gt;可以省略。使用&lt;optional_tag&gt;的目的是给出一些简短的注释。</span></span>
<span class="line"><span>例如，对于文件</span></span>
<span class="line"><span>genome.wustl.edu_OV.IlluminaGA_DNASeq.Level_2.7.6.0.tar.gz</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p>有效的MAF名称的为：</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>genome.wustl.edu_OV.IlluminaGA_DNASeq.Level_2.7.preliminary.somatic.maf</span></span>
<span class="line"><span>genome.wustl.edu_OV.IlluminaGA_DNASeq.Level_2.7.protected.maf</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div></div></div><h2 id="wiggle、bigwig和bedgraph" tabindex="-1"><a class="header-anchor" href="#wiggle、bigwig和bedgraph"><span>Wiggle、BigWig和bedgraph</span></a></h2><p><strong>简介</strong></p><p>Wiggle、BigWig和bedgraph仅仅用于追踪参考基因组的各个区域的覆盖度，测序深度。与sam/bam格式文件不同，bam或者bed格式的文件主要是为了追踪我们的reads到底比对到了参考基因组的哪些区域。注意这几者的差别。 Wiggle、BigWig和bedgraph均由UCSC规定的文件格式，可以无缝连接到UCSC的Genome Browser工具里面进行可视化。 wig和bigWig文件的优势在于可以体现出数据大小的变化和高低，例如组蛋白修饰的峰值等。</p><p><strong>一、定义</strong></p><p>**Bigwig：**简写为bw，它规定了全基因组数据的每个坐标区间的测序深度，bigWig是通过wig格式的文件转换的二进制压缩文件，是一种全基因组计算或实验信号数据的压缩的，索引的二进制格式，使用该格式更加节省空间。 <strong>Wiggle</strong>：简写为wig，表示基因组上一个区域的信号，可以上传至UCSC上进行可视化。而wig文件也是和BED文件类似的包含区域信息的文件，一般使用MACS峰值探测后可以产生wig格式的文件。</p><p><strong>二、Wiggle文件示例</strong></p><p>Wig是一种比较老的格式，展示连续值的数据，比如GC百分比，转录组数据等，Wig的数据被压缩的，储存在128个独特的仓中，当导出为其他格式时，会有很小的损失。</p><p>在UCSC中具体使用哪种数据格式，细节见：http://genomewiki.ucsc.edu/index.php/Selecting_a_graphing_track_data_format。以UCSC给的Wig参考文件为例，Wig的数据是面向行的，第一行定义了track的属性，比如track type=wiggle_0，指定track为Wig track，为默认选项。其余为可选。一般不用管这些参数，除非你已经很熟悉UCSC的Genome Browser工具。下面是wiggle文件示例。</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>track type=wiggle_0 name=&quot;variableStep&quot; description=&quot;variableStep format&quot; visibility=full autoScale=off viewLimits=0.0:25.0 color=50,150,255 yLineMark=11.76 yLineOnOff=on priority=10</span></span>
<span class="line"><span>variableStep chrom=chr19 span=150</span></span>
<span class="line"><span>49304701 10.0</span></span>
<span class="line"><span>49304901 12.5</span></span>
<span class="line"><span>49305401 15.0</span></span>
<span class="line"><span>49305601 17.5</span></span>
<span class="line"><span>49305901 20.0</span></span>
<span class="line"><span>49306081 17.5</span></span>
<span class="line"><span>49306301 15.0</span></span>
<span class="line"><span>49306691 12.5</span></span>
<span class="line"><span>49307871 10.0</span></span>
<span class="line"><span>#	200 base wide points graph at every 300 bases, 50 pixel high graph</span></span>
<span class="line"><span>#	autoScale off and viewing range set to [0:1000]</span></span>
<span class="line"><span>#	priority = 20 positions this as the second graph</span></span>
<span class="line"><span>#	Note, one-relative coordinate system in use for this format</span></span>
<span class="line"><span>track type=wiggle_0 name=&quot;fixedStep&quot; description=&quot;fixedStep format&quot; visibility=full autoScale=off viewLimits=0:1000 color=0,200,100 maxHeightPixels=100:50:20 graphType=points priority=20</span></span>
<span class="line"><span>fixedStep chrom=chr19 start=49307401 step=300 span=200</span></span>
<span class="line"><span>1000</span></span>
<span class="line"><span> 900</span></span>
<span class="line"><span> 800</span></span>
<span class="line"><span> 700</span></span>
<span class="line"><span> 600</span></span>
<span class="line"><span> 500</span></span>
<span class="line"><span> 400</span></span>
<span class="line"><span> 300</span></span>
<span class="line"><span> 200</span></span>
<span class="line"><span> 100</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p><strong>三、Wig文件详解</strong></p><p>Wig文件主要由两部分格式组成，**variableStep format*<em>和*<em>fixedStep format</em></em>。<strong>variableStep format</strong>以一个声明开始，明确了染色体的序号，跨度(span)。后面跟两列数据，染色体开始的碱基位置，数据的值value（可以理解为覆盖度）。span参数可以将含有相同value的连续碱基包含在一起，使数据更加简洁。如下，variableStep format span=150，包含的第一行数据49304701 10.0表示49304701-49304851有相同的value，为10.0。</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>variableStep chrom=chr2</span></span>
<span class="line"><span>300701 12.5</span></span>
<span class="line"><span>300702 12.5</span></span>
<span class="line"><span>300703 12.5</span></span>
<span class="line"><span>300704 12.5</span></span>
<span class="line"><span>300705 12.5</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p>等价于：</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>variableStep chrom=chr2 span=5</span></span>
<span class="line"><span>300701 12.5</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div></div></div><p>都表示在2号染色体300701-300705位置上有相同的value，且value=12.5。</p><p>第二部分为<strong>fixedStep format，</strong> 由声明和单列数据组成。声明部分和variableStep format中各变量的意义一样。</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>fixedStep chrom=chr3 start=400601 step=100 span=5</span></span>
<span class="line"><span>11</span></span>
<span class="line"><span>22</span></span>
<span class="line"><span>33</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p>表示在3号染色体400601-400605, 400701-400705, and 400801-400805三个区域，value值分别为11, 22, 和 33。Wig中的value值可以是整数，实数，正数或者负数。只有指定的位置有value值，没有制定的位置则没有value，且不会在UCSU Genome Browser中作出图。</p><p>其实对我们的bam文件，用samtools软件也可以很容易得到基因组区域的覆盖度和测序深度，比如：</p><p>samtools depth -r <strong>chr12:126073855-126073965</strong> Ip.sorted.bam</p><blockquote><p>chr12 126073855 5</p><p>chr12 126073856 15</p><p>chr12 126073857 31</p><p>chr12 126073858 40</p><p>chr12 126073859 44</p><p>chr12 126073860 52</p><p>...其余省略输出...</p></blockquote><p>这其实就是wig文件的雏形，但是wig文件会更复杂一点！</p><p>首先它不需要第一列了，因为全部是重复字段，只需要在每个染色体的第一行定义好染色体即可。</p><p><strong>四、文件转换示例</strong></p><p>BigWig格式是wig格式文件的二进制压缩版本，用于密集连续的数据，并在基因组浏览器中进行可视化，是UCSC推荐的一种格式。BigWig文件是由原始的Wig格式通过wigToBigWig工具转换过来， 转换工具命令示例：</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span># create the chrom.sizes file for the UCSC database (e.g., hg19).</span></span>
<span class="line"><span>fetchChromSizes  hg19 &gt; chrSize.txt</span></span>
<span class="line"><span># Convert wig to big wig:  </span></span>
<span class="line"><span>wigToBigWig input.wig chrSize.txt myBigWig.bw</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p>输出的BigWig文件大约比原始Wig文件多占用50%的内存（UCSC数据格式转换代码见：http://barcwiki.wi.mit.edu/wiki/SOPs/coordinates）。也可以由bedGraph 格式通过bedGraphToBigWig转换，这里不作展开。BigWig的主要优点是仅将显示特定区域所需的部分传输到UCSC，所以对于大数据集的文件，BigWig比Wig文件速度快。BigWig主要存在用户可以获得的网页上(http, https or ftp)，而不在UCSC服务器上，只有当前正在查看的染色体位置所需的文件部分在本地缓存为一个“稀疏文件”。</p><p><strong>五、BedGraph文件示例</strong></p><p>最后顺便讲讲BedGraph ，它的trace type和Wig文件很像，不过后面的数据和bed文件很类似，ChIPseq数据做完peak calling后的bed文件最短只有三列，染色体序号，染色体起始位置和结束位置。如下图，前面的声明和Wig类似，后面的四列分别表示染色体序号，起始位置，结束位置和value值。相当于为bed文件的延伸格式。</p><div class="language-text line-numbers-mode" data-ext="text" data-title="text"><pre class="shiki shiki-themes vitesse-light vitesse-dark vp-code" v-pre=""><code><span class="line"><span>browser position chr19:49302001-49304701</span></span>
<span class="line"><span>browser hide all</span></span>
<span class="line"><span>browser pack refGene encodeRegions</span></span>
<span class="line"><span>browser full altGraph</span></span>
<span class="line"><span>#	300 base wide bar graph, autoScale is on by default == graphing</span></span>
<span class="line"><span>#	limits will dynamically change to always show full range of data</span></span>
<span class="line"><span>#	in viewing window, priority = 20 positions this as the second graph</span></span>
<span class="line"><span>#	Note, zero-relative, half-open coordinate system in use for bedGraph format</span></span>
<span class="line"><span>track type=bedGraph name=&quot;BedGraph Format&quot; description=&quot;BedGraph format&quot; visibility=full color=200,100,0 altColor=0,100,200 priority=20</span></span>
<span class="line"><span>chr19 49302000 49302300 -1.0</span></span>
<span class="line"><span>chr19 49302300 49302600 -0.75</span></span>
<span class="line"><span>chr19 49302600 49302900 -0.50</span></span>
<span class="line"><span>chr19 49302900 49303200 -0.25</span></span>
<span class="line"><span>chr19 49303200 49303500 0.0</span></span>
<span class="line"><span>chr19 49303500 49303800 0.25</span></span>
<span class="line"><span>chr19 49303800 49304100 0.50</span></span>
<span class="line"><span>chr19 49304100 49304400 0.75</span></span>
<span class="line"><span>chr19 49304400 49304700 1.00</span></span></code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p><strong>文件格式选择</strong></p><p>Wig数据的元素大小必须是一样的。</p><p>如果数据大小不一样，应该使用bedGraph格式，如果数据过大，就转换为bigWig。</p></div><!----></div><footer class="page-footer" data-v-5c470ced data-v-13de951c><div class="edit-info" data-v-13de951c><div class="edit-link" data-v-13de951c><a class="auto-link link edit-link-button" href="https://github.com/Moonerss/blog/edit/main/docs/notes/NGS/filetype.md" target="_blank" rel="noreferrer" data-v-13de951c data-v-66dcefcc><!--[--><span class="vpi-square-pen edit-link-icon" aria-label="edit icon" data-v-13de951c></span> 在 GitHub 上编辑此页<!--]--><!----></a></div><!----></div><!----><nav class="prev-next" data-v-13de951c><div class="pager" data-v-13de951c><!----></div><div class="pager" data-v-13de951c><a class="auto-link link pager-link next" href="/NGS/concept/" data-v-13de951c data-v-66dcefcc><!--[--><span class="desc" data-v-13de951c>下一页</span><span class="title" data-v-13de951c>基础概念</span><!--]--><!----></a></div></nav></footer><!----><!--]--></div></div></div></div><button style="display:none;" type="button" class="back-to-top-button" aria-label="back to top" data-v-0f4179d5 data-v-0af269ee><span class="percent" data-v-0af269ee>0%</span><span class="show icon vpi-back-to-top" data-v-0af269ee></span><svg aria-hidden="true" data-v-0af269ee><circle cx="50%" cy="50%" style="stroke-dasharray:calc(0% - 12.566370614359172px) calc(314.1592653589793% - 12.566370614359172px);" data-v-0af269ee></circle></svg></button><footer class="plume-footer has-sidebar" data-v-0f4179d5 data-v-24205ac6><div class="container" data-v-24205ac6><p class="message" data-v-24205ac6>Power by <a target="_blank" href="https://v2.vuepress.vuejs.org/">VuePress</a> & <a target="_blank" href="https://github.com/pengzhanbo/vuepress-theme-plume">vuepress-theme-plume</a></p><p class="copyright" data-v-24205ac6>Copyright © 2024-present Jeason</p></div></footer><!--]--></div><!--]--></div><!--[--><!--]--><!--]--></div><script type="module" src="/assets/app-B6iXF0Jq.js" defer></script></body></html>