import{_ as i,o as s,c as a,e as t,a as e,b as n}from"./app-B6iXF0Jq.js";const o={},g=t('<h2 id="eggnog-mapper介绍" tabindex="-1"><a class="header-anchor" href="#eggnog-mapper介绍"><span>eggNOG-Mapper介绍</span></a></h2><p>通常功能注释的思路都是基于序列相似性找直系同源基因，常见的方法就是BLAST+BLAST2GO， 或者是InterProScan。eggNOG-mapper的作者认为这种方法不够可靠，毕竟你有可能找到的的是旁系同源基因。近期对多个工具的整体评估发现eggNOG(evolutionary genealogy of genes: Non-supervised Orthologous Groups)在区分旁系同源基因和直系同源基因上表现不错，因此基于eggNOG数据库开发了eggNOG-mapper工具，用于对新序列进行功能注释。</p><p>eggNOG-mapper的算法实现如下：</p><ol><li>序列比对。首先，每条蛋白序列用HMMER3在整理的eggNOG数据库中搜索。由于每个HMM匹配都和一个功能注释的eggNOG OG对应，这一步就提供了初步的注释信息,</li><li>每条蛋白序列用phmmer在最佳匹配的HMM对应的一组eggNOG蛋白中进一步搜索,</li><li>每条序列的最佳匹配结果以 <strong>seed ortholog</strong> 形式存放，用于获取其他直系同源基因。</li></ol><p>目前eggNOG HMM数据库中拥有1,911,745个OG，覆盖了1,678种细菌,115种古细菌，238种真核物种以及352种病毒。除了HMMER3外，还而可用DIAMOND直接对所有的eggNOG蛋白序列进行搜索，它的速度更快，适合类似于宏基因组这类大数据集，或者是已有物种和eggNOG所收集的物种比较近</p><div style="text-align:center;"><p><img src="https://cdn.jsdelivr.net/gh/Moonerss/CDN/paper/eggnog/Pasted_image_20231124191335.png" alt=""></p></div><ol start="4"><li>推测直系同源基因。每个用于检索的蛋白序列的最佳匹配序列会对应eggNOG的一个蛋白, 这些蛋白基于预分析的eggNOG进化树数据库会提取一组更加精细的直系同源基因。这一步还会根据bit-screo或E-value对结果进行一次过来，剔除同源性不高的结果</li></ol><div style="text-align:center;"><p><img src="https://cdn.jsdelivr.net/gh/Moonerss/CDN/paper/eggnog/Pasted_image_20231124191355.png" alt=""></p></div><ol start="5"><li>功能注释。用于搜索的蛋白序列对应的直系同源基因的功能描述就是最终的注释结果。比如说GO, KEGG, COG等。</li></ol><div style="text-align:center;"><p><img src="https://cdn.jsdelivr.net/gh/Moonerss/CDN/paper/eggnog/Pasted_image_20231124191407.png" alt=""></p></div><h2 id="安装" tabindex="-1"><a class="header-anchor" href="#安装"><span>安装</span></a></h2><p>安装本体eggnog-mapper之前，需要先保证服务器上已经安装Python3.7(BioPython模块), wget, HMMER3 和/或 DIAMON，此外你还得保证70G的空间用于存放注释数据库和FASTA文件，对于真核生物至少保证90G的服务器内存。</p>',12),r=e("div",{class:"language-bash line-numbers-mode","data-ext":"sh","data-title":"sh"},[e("pre",{class:"shiki shiki-themes vitesse-light vitesse-dark vp-code","v-pre":""},[e("code",null,[e("span",{class:"line"},[e("span",{style:{"--shiki-light":"#59873A","--shiki-dark":"#80A665"}},"git"),e("span",{style:{"--shiki-light":"#B56959","--shiki-dark":"#C98A7D"}}," clone"),e("span",{style:{"--shiki-light":"#B56959","--shiki-dark":"#C98A7D"}}," https://github.com/jhcepas/eggnog-mapper.git")])])]),e("div",{class:"line-numbers","aria-hidden":"true"},[e("div",{class:"line-number"})])],-1),p=e("p",null,"之后需要下载所需要的数据库。eggNOG提供了107个不同物种的HMM数据库(xxxNOG)以及三个优化数据库, euk对应真核生物，bact对应细菌, arch对应古细菌(Archeabacteria), 以及一个病毒数据库(viruses). 这三个优化数据库包含了属于该分类内的所有物种的HMM模型。",-1),l=e("div",{class:"language-bash line-numbers-mode","data-ext":"sh","data-title":"sh"},[e("pre",{class:"shiki shiki-themes vitesse-light vitesse-dark vp-code","v-pre":""},[e("code",null,[e("span",{class:"line"},[e("span",{style:{"--shiki-light":"#998418","--shiki-dark":"#B8A965"}},"cd"),e("span",{style:{"--shiki-light":"#B56959","--shiki-dark":"#C98A7D"}}," eggnog-mapper")]),n(`
`),e("span",{class:"line"},[e("span",{style:{"--shiki-light":"#59873A","--shiki-dark":"#80A665"}},"python3"),e("span",{style:{"--shiki-light":"#B56959","--shiki-dark":"#C98A7D"}}," ./download_eggnog_data.py")])])]),e("div",{class:"line-numbers","aria-hidden":"true"},[e("div",{class:"line-number"}),e("div",{class:"line-number"})])],-1),h=e("h2",{id:"用法",tabindex:"-1"},[e("a",{class:"header-anchor",href:"#用法"},[e("span",null,"用法")])],-1),d=e("p",null,"eggnog-mapper用起来非常的简单，你需要提供蛋白序列或基因组序列等作为输入",-1),c=e("div",{class:"language-bash line-numbers-mode","data-ext":"sh","data-title":"sh"},[e("pre",{class:"shiki shiki-themes vitesse-light vitesse-dark vp-code","v-pre":""},[e("code",null,[e("span",{class:"line"},[e("span",{style:{"--shiki-light":"#59873A","--shiki-dark":"#80A665"}},"python3"),e("span",{style:{"--shiki-light":"#B56959","--shiki-dark":"#C98A7D"}}," path/to/emapper.py"),e("span",{style:{"--shiki-light":"#A65E2B","--shiki-dark":"#C99076"}}," -i"),e("span",{style:{"--shiki-light":"#B56959","--shiki-dark":"#C98A7D"}}," test.fasta"),e("span",{style:{"--shiki-light":"#A65E2B","--shiki-dark":"#C99076"}}," --itype"),e("span",{style:{"--shiki-light":"#B56959","--shiki-dark":"#C98A7D"}}," genome"),e("span",{style:{"--shiki-light":"#A65E2B","--shiki-dark":"#C99076"}}," --cpu"),e("span",{style:{"--shiki-light":"#2F798A","--shiki-dark":"#4C9A91"}}," 10"),e("span",{style:{"--shiki-light":"#A65E2B","--shiki-dark":"#C99076"}}," -o"),e("span",{style:{"--shiki-light":"#B56959","--shiki-dark":"#C98A7D"}}," rapa")])])]),e("div",{class:"line-numbers","aria-hidden":"true"},[e("div",{class:"line-number"})])],-1),m=t('<p>eggnog-mapper默认是以HMMER进行序列搜索，可以通过-m diamond更改成DIAMOND,参数一般使用默认就好。</p><p><code>--itype</code>表示输入的序列文件类型，可选的有：CDS, proteins, genome, metagenome</p><p><code>--cpu</code>选取线程数目</p><p><code>-o</code>表示输出文件的前缀，默认输出在当前文件夹下</p><h2 id="结果" tabindex="-1"><a class="header-anchor" href="#结果"><span>结果</span></a></h2><p>eggnog-mapper会生成三个文件:</p><ul><li>[project_name].emapper.hmm_hits: 记录每个用于搜索序列对应的所有的显著性的eggNOG Orthologous Groups(OG). 所有标记为&quot;-&quot;则表明该序列未找到可能的OG</li><li>[project_name].emapper.seed_orthologs: 记录每个用于搜索序列对的的最佳的OG, 也就是[project_name].emapper.hmm_hits里选择得分最高的结果。之后会从eggNOG中提取更精细的直系同源关系(orthology relationships)</li><li>[project_name].emapper.annotations: 该文件提供了最终的注释结果。大部分需要的内容都可以通过写脚本从从提取，一共有13列。</li></ul><p>[project_name].emapper.annotations每一列对应的记录如下：</p><ol><li>query_name: 检索的基因名或者其他ID</li><li>seed_ortholog: eggNOG中最佳的蛋白匹配</li><li>evalue: 最佳匹配的e-value</li><li>score: 最佳匹配的bit-score</li><li>eggNOG_OGs: 预测的基因名，特别指的是类似AP2有一定含义的基因名，而不是AT2G17950这类编号</li></ol>',9),k=[g,r,p,l,h,d,c,m];function u(_,y){return s(),a("div",null,k)}const G=i(o,[["render",u],["__file","index.html.vue"]]),N=JSON.parse(`{"path":"/bioinfo/eggnog/","title":"eggnog-mapper功能注释","lang":"zh-CN","frontmatter":{"title":"eggnog-mapper功能注释","author":"Jeason","createTime":"2023/04/05 22:47:45","permalink":"/bioinfo/eggnog/","tags":["biosoftware"],"description":"eggNOG-Mapper介绍 通常功能注释的思路都是基于序列相似性找直系同源基因，常见的方法就是BLAST+BLAST2GO， 或者是InterProScan。eggNOG-mapper的作者认为这种方法不够可靠，毕竟你有可能找到的的是旁系同源基因。近期对多个工具的整体评估发现eggNOG(evolutionary genealogy of gene...","head":[["script",{"id":"check-dark-mode"},";(function () {const um= localStorage.getItem('vuepress-theme-appearance') || 'auto';const sm = window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches;if (um === 'dark' || (um !== 'light' && sm)) {document.documentElement.classList.add('dark');}})();"],["script",{"id":"check-mac-os"},"document.documentElement.classList.toggle('mac', /Mac|iPhone|iPod|iPad/i.test(navigator.platform))"],["meta",{"property":"og:url","content":"https://jeason.netlify.app/bioinfo/eggnog/"}],["meta",{"property":"og:title","content":"eggnog-mapper功能注释"}],["meta",{"property":"og:description","content":"eggNOG-Mapper介绍 通常功能注释的思路都是基于序列相似性找直系同源基因，常见的方法就是BLAST+BLAST2GO， 或者是InterProScan。eggNOG-mapper的作者认为这种方法不够可靠，毕竟你有可能找到的的是旁系同源基因。近期对多个工具的整体评估发现eggNOG(evolutionary genealogy of gene..."}],["meta",{"property":"og:type","content":"article"}],["meta",{"property":"og:image","content":"https://cdn.jsdelivr.net/gh/Moonerss/CDN/paper/eggnog/Pasted_image_20231124191335.png"}],["meta",{"property":"og:locale","content":"zh-CN"}],["meta",{"property":"og:updated_time","content":"2024-05-20T06:59:19.000Z"}],["meta",{"property":"article:author","content":"Jeason"}],["meta",{"property":"article:tag","content":"biosoftware"}],["meta",{"property":"article:modified_time","content":"2024-05-20T06:59:19.000Z"}],["script",{"type":"application/ld+json"},"{\\"@context\\":\\"https://schema.org\\",\\"@type\\":\\"Article\\",\\"headline\\":\\"eggnog-mapper功能注释\\",\\"image\\":[\\"https://cdn.jsdelivr.net/gh/Moonerss/CDN/paper/eggnog/Pasted_image_20231124191335.png\\",\\"https://cdn.jsdelivr.net/gh/Moonerss/CDN/paper/eggnog/Pasted_image_20231124191355.png\\",\\"https://cdn.jsdelivr.net/gh/Moonerss/CDN/paper/eggnog/Pasted_image_20231124191407.png\\"],\\"dateModified\\":\\"2024-05-20T06:59:19.000Z\\",\\"author\\":[{\\"@type\\":\\"Person\\",\\"name\\":\\"Jeason\\"}]}"]]},"headers":[{"level":2,"title":"eggNOG-Mapper介绍","slug":"eggnog-mapper介绍","link":"#eggnog-mapper介绍","children":[]},{"level":2,"title":"安装","slug":"安装","link":"#安装","children":[]},{"level":2,"title":"用法","slug":"用法","link":"#用法","children":[]},{"level":2,"title":"结果","slug":"结果","link":"#结果","children":[]}],"isBlogPost":true,"readingTime":{"minutes":3.63,"words":1090},"git":{"updatedTime":1716188359000},"autoDesc":true,"filePathRelative":"bioinfo/eggnog.md","categoryList":[{"type":10001,"name":"bioinfo"}]}`);export{G as comp,N as data};
